--- Python-3.7.2/Modules/mmapmodule.c	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.32/Modules/mmapmodule.c	2019-01-05 15:52:35.709483115 +0100
@@ -79,6 +79,12 @@
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
+
 typedef enum
 {
     ACCESS_DEFAULT,
