--- Python-3.7.2/Modules/signalmodule.c	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.32/Modules/signalmodule.c	2019-01-05 15:52:35.717482852 +0100
@@ -40,6 +40,13 @@
 #define SIG_ERR ((PyOS_sighandler_t)(-1))
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    #define SIGRTMIN 32
+    #define SIGRTMAX _NSIG
+#endif
+//PMPP API<21
+
 #ifndef NSIG
 # if defined(_NSIG)
 #  define NSIG _NSIG            /* For BSD/SysV */
