--- Python-3.7.2/Modules/timemodule.c	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.32/Modules/timemodule.c	2019-01-05 15:52:35.721482722 +0100
@@ -477,6 +477,7 @@
 }
 
 #ifndef HAVE_TIMEGM
+#ifndef CRYSTAX
 static time_t
 timegm(struct tm *p)
 {
@@ -488,6 +489,7 @@
         ((p->tm_year-1)/100)*86400 + ((p->tm_year+299)/400)*86400;
 }
 #endif
+#endif
 
 PyDoc_STRVAR(gmtime_doc,
 "gmtime([seconds]) -> (tm_year, tm_mon, tm_mday, tm_hour, tm_min,\n\
