--- Python-3.7.2/Modules/_localemodule.c	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.32/Modules/_localemodule.c	2019-01-05 15:52:35.701483380 +0100
@@ -96,6 +96,10 @@
         PyErr_SetString(Error, "invalid locale category");
         return NULL;
     }
+#else
+#ifdef __ANDROID__
+    return PyUnicode_FromFormat("%s", "C");
+#endif
 #endif
 
     if (locale) {
@@ -213,7 +217,16 @@
     }
 
     /* if LC_NUMERIC is different in the C library, use saved value */
-    l = localeconv();
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    /* Don't even try on Android's broken locale.h. */
+    /* maybe can use c++ stdlib to implement */
+    goto failed;
+#else
+     /* if LC_NUMERIC is different in the C library, use saved value */
+     l = localeconv(); //PATCHED
+#endif
+//PMPP API<21
 
     /* hopefully, the localeconv result survives the C library calls
        involved herein */
