--- Python-3.7.2/Python/pystrtod.c	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.32/Python/pystrtod.c	2019-01-05 15:52:35.741482064 +0100
@@ -176,9 +176,14 @@
     assert(nptr != NULL);
 
     fail_pos = NULL;
-
-    locale_data = localeconv();
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    decimal_point = ".";
+#else
+    locale_data = localeconv(); //PATCHED
     decimal_point = locale_data->decimal_point;
+#endif
+//PMPP API<21
     decimal_point_len = strlen(decimal_point);
 
     assert(decimal_point_len != 0);
