--- Python-3.8.0b4/Include/internal/pycore_pylifecycle.h	2019-08-29 23:59:20.000000000 +0200
+++ Python-3.8.0b4.em/Include/internal/pycore_pylifecycle.h	2019-09-18 13:13:27.994783283 +0200
@@ -83,7 +83,7 @@
 extern void PyLong_Fini(void);
 extern void _PyFaulthandler_Fini(void);
 extern void _PyHash_Fini(void);
-extern int _PyTraceMalloc_Fini(void);
+extern void _PyTraceMalloc_Fini(void);
 extern void _PyWarnings_Fini(PyInterpreterState *interp);
 
 extern void _PyGILState_Init(
