--- Python-3.8.0b4/Modules/posixmodule.c	2019-08-29 23:59:20.000000000 +0200
+++ Python-3.8.0b4.em/Modules/posixmodule.c	2019-09-17 20:08:46.036508906 +0200
@@ -5303,9 +5303,13 @@
 
    if (setsigmask) {
         sigset_t set;
+		#if __EMSCRIPTEN__
+			#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+		#else
         if (!_Py_Sigset_Converter(setsigmask, &set)) {
             goto fail;
         }
+		#endif
         errno = posix_spawnattr_setsigmask(attrp, &set);
         if (errno) {
             posix_error();
@@ -5316,9 +5320,13 @@
 
     if (setsigdef) {
         sigset_t set;
+		#if __EMSCRIPTEN__
+			#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+		#else
         if (!_Py_Sigset_Converter(setsigdef, &set)) {
             goto fail;
         }
+		#endif
         errno = posix_spawnattr_setsigdefault(attrp, &set);
         if (errno) {
             posix_error();
