--- panda3d-master/dtool/src/interrogatedb/py_panda.h	2019-08-26 10:07:58.000000000 +0200
+++ panda3d-master.32/dtool/src/interrogatedb/py_panda.h	2019-09-01 19:10:50.991187599 +0200
@@ -210,7 +210,7 @@
 #endif
 
 EXPCL_PYPANDA PyObject *Dtool_Raise_AssertionError();
-EXPCL_PYPANDA PyObject *Dtool_Raise_TypeError(const char *message);
+EXPCL_INTERROGATEDB PyObject *Dtool_Raise_TypeError(const char *message);
 EXPCL_PYPANDA PyObject *Dtool_Raise_ArgTypeError(PyObject *obj, int param, const char *function_name, const char *type_name);
 EXPCL_PYPANDA PyObject *Dtool_Raise_AttributeError(PyObject *obj, const char *attribute);
 
