--- panda3d-master/panda/src/android/PandaActivity.java	2019-08-26 10:07:58.000000000 +0200
+++ panda3d-master.32/panda/src/android/PandaActivity.java	2019-09-01 19:10:51.019186589 +0200
@@ -12,6 +12,7 @@
  */
 
 package org.panda3d.android;
+/*
 
 import android.app.NativeActivity;
 import android.content.Intent;
@@ -22,10 +23,6 @@
 import org.panda3d.android.NativeIStream;
 import org.panda3d.android.NativeOStream;
 
-/**
- * The entry point for a Panda-based activity.  Loads the Panda libraries and
- * also provides some utility functions.
- */
 public class PandaActivity extends NativeActivity {
     private static final Bitmap.Config sConfigs[] = {
             null,
@@ -118,3 +115,4 @@
         System.loadLibrary("pandagles");
     }
 }
+*/
