--- Python-3.9.0/Lib/os.py	2020-10-05 17:07:58.000000000 +0200
+++ Python-3.9.0-aosp/Lib/os.py	2020-11-09 08:45:24.029786771 +0100
@@ -1114,3 +1114,12 @@
             cookie,
             nt._remove_dll_directory
         )
+
+try:
+    getpid
+except:
+    import sys
+    print("210: os.getpid() is broken",file=sys.stderr)
+    def getpid():
+        return int(open('/proc/self/stat').read().split(' ')[0])
+
