--- Python-3.9.0/Lib/modulefinder.py	2020-10-05 17:07:58.000000000 +0200
+++ Python-3.9.0-aosp/Lib/modulefinder.py	2020-11-09 08:45:24.009787462 +0100
@@ -237,8 +237,8 @@
             if q:
                 self.msgout(4, "find_head_package ->", (q, tail))
                 return q, tail
-        self.msgout(4, "raise ImportError: No module named", qname)
-        raise ImportError("No module named " + qname)
+        self.msgout(4, "raise ImportError: No module named1", qname)
+        raise ImportError("No module named1 " + qname)
 
     def load_tail(self, q, tail):
         self.msgin(4, "load_tail", q, tail)
@@ -251,7 +251,7 @@
             m = self.import_module(head, mname, m)
             if not m:
                 self.msgout(4, "raise ImportError: No module named", mname)
-                raise ImportError("No module named " + mname)
+                raise ImportError("No module named2 " + mname)
         self.msgout(4, "load_tail ->", m)
         return m
 
@@ -267,7 +267,7 @@
                 subname = "%s.%s" % (m.__name__, sub)
                 submod = self.import_module(sub, subname, m)
                 if not submod:
-                    raise ImportError("No module named " + subname)
+                    raise ImportError("No module named3 " + subname)
 
     def find_all_submodules(self, m):
         if not m.__path__:
