--- Python-3.9.0/Modules/timemodule.c	2020-10-05 17:07:58.000000000 +0200
+++ Python-3.9.0-aosp/Modules/timemodule.c	2020-11-09 08:45:24.109784008 +0100
@@ -464,6 +464,7 @@
 }
 
 #ifndef HAVE_TIMEGM
+#ifndef CRYSTAX
 static time_t
 timegm(struct tm *p)
 {
@@ -475,6 +476,7 @@
         ((p->tm_year-1)/100)*86400 + ((p->tm_year+299)/400)*86400;
 }
 #endif
+#endif
 
 PyDoc_STRVAR(gmtime_doc,
 "gmtime([seconds]) -> (tm_year, tm_mon, tm_mday, tm_hour, tm_min,\n\
