--- Python-3.9.0/Modules/readline.c	2020-10-05 17:07:58.000000000 +0200
+++ Python-3.9.0-aosp/Modules/readline.c	2020-11-09 08:45:24.097784422 +0100
@@ -1186,7 +1186,9 @@
         int has_input = 0, err = 0;
 
         while (!has_input)
-        {               struct timeval timeout = {0, 100000}; /* 0.1 seconds */
+        {
+
+            struct timeval timeout = {0, 10000}; /* 0.01 seconds */
 
             /* [Bug #1552726] Only limit the pause if an input hook has been
                defined.  */
