--- Python-3.9.0/Parser/tokenizer.c	2020-10-05 17:07:58.000000000 +0200
+++ Python-3.9.0-wasm/Parser/tokenizer.c	2020-11-09 08:40:31.539884469 +0100
@@ -849,7 +849,7 @@
             return Py_CHARMASK(*tok->cur++);
         }
         if (tok->prompt != NULL) {
-            char *newtok = PyOS_Readline(stdin, stdout, tok->prompt);
+            char *newtok = PyOS_Readline(tok->fp? tok->fp : stdin, stdout, tok->prompt);
             if (newtok != NULL) {
                 char *translated = translate_newlines(newtok, 0, tok);
                 PyMem_FREE(newtok);
