--- cpython-3.8.3/Modules/_localemodule.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-aosp/Modules/_localemodule.c	2020-06-07 10:47:21.601947816 +0200
@@ -97,6 +97,10 @@
         PyErr_SetString(Error, "invalid locale category");
         return NULL;
     }
+#else
+#ifdef __ANDROID__
+    return PyUnicode_FromFormat("%s", "C");
+#endif
 #endif
 
     if (locale) {
@@ -214,7 +218,16 @@
     }
 
     /* if LC_NUMERIC is different in the C library, use saved value */
-    lc = localeconv();
+//PMPP API<21
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+    /* Don't even try on Android's broken locale.h. */
+    /* maybe can use c++ stdlib to implement */
+     lc = broken_localeconv();
+#else
+     /* if LC_NUMERIC is different in the C library, use saved value */
+     lc = localeconv(); //PATCHED
+#endif
+//PMPP API<21
 
     /* hopefully, the localeconv result survives the C library calls
        involved herein */
