--- cpython-3.8.3/Modules/mmapmodule.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-aosp/Modules/mmapmodule.c	2020-06-07 10:47:21.603947787 +0200
@@ -79,6 +79,12 @@
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
+
 typedef enum
 {
     ACCESS_DEFAULT,
