--- cpython-3.8.3/Python/import.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-aosp/Python/import.c	2020-06-07 10:47:21.615947613 +0200
@@ -2198,7 +2198,9 @@
         Py_DECREF(name);
         return NULL;
     }
-
+#if __ANDROID__
+    printf("2151: name: %s, path:%s\n",PyUnicode_AsUTF8(name),PyUnicode_AsUTF8(path));
+#endif
     mod = _PyImport_FindExtensionObject(name, path);
     if (mod != NULL || PyErr_Occurred()) {
         Py_DECREF(name);
