--- cpython-3.8.3/Python/dynload_shlib.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-aosp/Python/dynload_shlib.c	2020-06-07 10:47:21.609947700 +0200
@@ -34,6 +34,9 @@
 */
 
 const char *_PyImport_DynLoadFiletab[] = {
+#if __ANDROID__
+    "." SOABI ".so",
+#else
 #ifdef __CYGWIN__
     ".dll",
 #else  /* !__CYGWIN__ */
@@ -44,6 +47,7 @@
     ".abi" PYTHON_ABI_STRING ".so",
     ".so",
 #endif  /* __CYGWIN__ */
+#endif  /* __ANDROID__ */
     NULL,
 };
 
