--- cpython-3.8.3/Python/formatter_unicode.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-aosp/Python/formatter_unicode.c	2020-06-07 10:47:21.613947642 +0200
@@ -690,6 +690,7 @@
 
 static const char no_grouping[1] = {CHAR_MAX};
 
+
 /* Find the decimal point character(s?), thousands_separator(s?), and
    grouping description, either for the current locale if type is
    LT_CURRENT_LOCALE, a hard-coded locale if LT_DEFAULT_LOCALE or
