--- cpython-3.8.3/Python/fileutils.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-aosp/Python/fileutils.c	2020-06-07 10:47:21.611947671 +0200
@@ -51,6 +51,16 @@
 }
 
 
+//PMPP API<21
+#ifdef __ANDROID__
+    #if __ANDROID_API__ < 21
+        #ifndef CRYSTAX
+            #define NO_LOCALECONV 1
+        #endif
+    #endif
+#endif
+//PMPP API<21
+
 PyObject *
 _Py_device_encoding(int fd)
 {
@@ -1976,3 +1986,5 @@
     PyMem_Free(oldloc);
     return res;
 }
+
+
