--- Python-3.7.0b4/config.sub	2018-05-02 09:41:45.000000000 +0200
+++ Python-3.7.0b4.em/config.sub	2018-05-19 09:14:37.522856174 +0200
@@ -118,7 +118,8 @@
   linux-musl* | linux-uclibc* | uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | \
   knetbsd*-gnu* | netbsd*-gnu* | netbsd*-eabi* | \
   kopensolaris*-gnu* | cloudabi*-eabi* | \
-  storm-chaos* | os2-emx* | rtmk-nova*)
+  storm-chaos* | os2-emx* | rtmk-nova* | \
+  emscripten)
     os=-$maybe_os
     basic_machine=`echo "$1" | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`
     ;;
@@ -250,6 +251,7 @@
 	| am33_2.0 \
 	| arc | arceb \
 	| arm | arm[bl]e | arme[lb] | armv[2-8] | armv[3-8][lb] | armv6m | armv[78][arm] \
+	| asmjs \
 	| avr | avr32 \
 	| ba \
 	| be32 | be64 \
@@ -1522,6 +1524,8 @@
 			;;
 		esac
 		;;
+	-emscripten)
+		;;
 	-nacl*)
 		;;
 	-ios)
