--- Python-3.7.0b4/Python/fileutils.c	2018-05-02 09:41:45.000000000 +0200
+++ Python-3.7.0b4.em/Python/fileutils.c	2018-05-19 09:13:14.150858105 +0200
@@ -964,6 +964,9 @@
 static int
 set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
 {
+#ifdef __EMSCRIPTEN__
+    return 0;
+#else
 #ifdef MS_WINDOWS
     HANDLE handle;
     DWORD flags;
@@ -1080,6 +1083,7 @@
     }
     return 0;
 #endif
+#endif
 }
 
 /* Make the file descriptor non-inheritable.
