--- Python-3.7.1/configure	2018-10-20 08:04:19.000000000 +0200
+++ Python-3.7.1.em/configure	2018-11-17 07:18:26.254871385 +0100
@@ -3253,6 +3253,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	asmjs-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3297,6 +3300,9 @@
 	*-*-cygwin*)
 		_host_cpu=
 		;;
+	asmjs-*-*)
+		_host_cpu=
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -9208,8 +9214,9 @@
 # -- usually .so, .sl on HP-UX, .dll on Cygwin
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking the extension of shared libraries" >&5
 $as_echo_n "checking the extension of shared libraries... " >&6; }
+
 if test -z "$SHLIB_SUFFIX"; then
-	case $ac_sys_system in
+    case $ac_sys_system in
 	hp*|HP*)
 		case `uname -m` in
 			ia64) SHLIB_SUFFIX=.so;;
@@ -9220,6 +9227,12 @@
 	*)	   SHLIB_SUFFIX=.so;;
 	esac
 fi
+
+case $ac_sys_system in
+Emscripten)
+    SHLIB_SUFFIX=.wasm;;
+esac
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $SHLIB_SUFFIX" >&5
 $as_echo "$SHLIB_SUFFIX" >&6; }

@@ -11538,6 +11551,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -11563,6 +11579,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -12212,7 +12231,9 @@
 #ifdef HAVE_GRP_H
 #include <grp.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error setgroups
+#endif
 int
 main ()
 {
@@ -12750,7 +12771,9 @@
 #else
 #include <sys/types.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error makedev
+#endif
 int
 main ()
 {
@@ -14968,10 +14991,14 @@


 case $ac_sys_system in
+    Emscripten)
+        SOABI="cpython-${VERSION}-emscripten"
+        EXT_SUFFIX=".wasm"
+        ;;
     Linux*|GNU*|Darwin)
-	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+	    EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
     *)
-	EXT_SUFFIX=${SHLIB_SUFFIX};;
+	    EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac

 { $as_echo "$as_me:${as_lineno-$LINENO}: checking LDVERSION" >&5
