--- Python-3.7.1/Modules/readline.c	2018-10-20 08:04:19.000000000 +0200
+++ Python-3.7.1.em/Modules/readline.c	2018-11-17 07:18:26.270870859 +0100
@@ -1185,6 +1185,11 @@
 
     completed_input_string = not_done_reading;
 
+    #ifdef __EMSCRIPTEN__
+        #error no no no i won't go there
+    #endif
+
+
     while (completed_input_string == not_done_reading) {
         int has_input = 0, err = 0;
 
