--- cpython-3.7.1rc1/Modules/readline.c	2018-09-26 08:16:09.000000000 +0200
+++ cpython-3.7.1rc1.em/Modules/readline.c	2018-10-16 00:25:40.667417701 +0200
@@ -1185,6 +1185,11 @@
 
     completed_input_string = not_done_reading;
 
+    #ifdef __EMSCRIPTEN__
+        #error no no no i won't go there
+    #endif
+
+
     while (completed_input_string == not_done_reading) {
         int has_input = 0, err = 0;
 
