--- Python-3.8.5/Modules/_ctypes/callbacks.c	2020-07-20 15:01:32.000000000 +0200
+++ python3-wasm/Modules/_ctypes/callbacks.c	2020-09-29 15:03:33.142131929 +0200
@@ -361,8 +361,8 @@
 
     assert(CThunk_CheckExact((PyObject *)p));
 
-    p->pcl_write = ffi_closure_alloc(sizeof(ffi_closure),
-                                                                         &p->pcl_exec);
+    p->pcl_write = ffi_closure_alloc(sizeof(ffi_closure), &p->pcl_exec);
+
     if (p->pcl_write == NULL) {
         PyErr_NoMemory();
         goto error;
