--- Python-3.7.2/configure	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.em/configure	2019-01-26 10:07:31.057633677 +0100
@@ -3261,6 +3261,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	asmjs-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3305,6 +3308,9 @@
 	*-*-cygwin*)
 		_host_cpu=
 		;;
+	asmjs-*-*)
+		_host_cpu=
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -9337,8 +9343,9 @@
 # -- usually .so, .sl on HP-UX, .dll on Cygwin
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking the extension of shared libraries" >&5
 $as_echo_n "checking the extension of shared libraries... " >&6; }
+
 if test -z "$SHLIB_SUFFIX"; then
-	case $ac_sys_system in
+    case $ac_sys_system in
 	hp*|HP*)
 		case `uname -m` in
 			ia64) SHLIB_SUFFIX=.so;;
@@ -9349,6 +9356,12 @@
 	*)	   SHLIB_SUFFIX=.so;;
 	esac
 fi
+
+case $ac_sys_system in
+Emscripten)
+    SHLIB_SUFFIX=.wasm;;
+esac
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $SHLIB_SUFFIX" >&5
 $as_echo "$SHLIB_SUFFIX" >&6; }
 
@@ -11705,6 +11718,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -11730,6 +11746,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -12379,7 +12398,9 @@
 #ifdef HAVE_GRP_H
 #include <grp.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error setgroups
+#endif
 int
 main ()
 {
@@ -12917,7 +12938,9 @@
 #else
 #include <sys/types.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error makedev
+#endif
 int
 main ()
 {
@@ -15135,10 +15158,14 @@
 
 
 case $ac_sys_system in
+    Emscripten)
+        SOABI="cpython-${VERSION}-emscripten"
+        EXT_SUFFIX=".wasm"
+        ;;
     Linux*|GNU*|Darwin)
-	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+	    EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
     *)
-	EXT_SUFFIX=${SHLIB_SUFFIX};;
+	    EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking LDVERSION" >&5
