--- Python-3.7.2/Lib/subprocess.py	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.em/Lib/subprocess.py	2019-01-26 10:07:31.065633385 +0100
@@ -40,10 +40,9 @@
 getstatusoutput(...): Runs a command in the shell, waits for it to complete,
     then returns a (exitcode, output) tuple
 """
-
 import sys
 _mswindows = (sys.platform == "win32")
-
+_emscripten = (sys.platform == "emscripten")
 import io
 import os
 import time
@@ -149,7 +148,8 @@
                                lpAttributeList=attr_list)
 
 else:
-    import _posixsubprocess
+    if not _emscripten:
+        import _posixsubprocess
     import select
     import selectors
     import threading
