--- Python-3.7.2/Lib/hashlib.py	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.em/Lib/hashlib.py	2019-01-26 10:07:31.061633531 +0100
@@ -55,10 +55,14 @@
 
 # This tuple and __get_builtin_constructor() must be modified if a new
 # always available algorithm is added.
-__always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512',
-                      'blake2b', 'blake2s',
-                      'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512',
-                      'shake_128', 'shake_256')
+import sys
+if sys.platform=='emscripten':
+    __always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')
+else:
+    __always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512',
+                          'blake2b', 'blake2s',
+                          'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512',
+                          'shake_128', 'shake_256')
 
 
 algorithms_guaranteed = set(__always_supported)
