--- Python-3.7.2/Parser/myreadline.c	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2.em/Parser/myreadline.c	2019-01-26 10:07:31.097632218 +0100
@@ -275,6 +275,11 @@
         return NULL;
     case -1: /* EOF */
     case -2: /* Error */
+    case -3:
+        #ifdef __EMSCRIPTEN__
+        #warning ------------- no no no i won't go there --------------
+        #endif
+        break;
     default: /* Shouldn't happen */
         *p = '\0';
         break;
