--- cpython-master/Modules/main.c	2018-01-23 22:26:07.000000000 +0100
+++ cpython-master.api19/Modules/main.c	2018-01-24 01:56:29.000000000 +0100
@@ -62,6 +62,14 @@
 #define WCSTOK wcstok
 #endif
 
+//PMPP API<21
+#ifdef __ANDROID__
+    #if __ANDROID_API__ < 21
+        #define NO_LOCALECONV 1
+    #endif
+#endif
+//PMPP API<21
+
 /* For Py_GetArgcArgv(); set by main() */
 static wchar_t **orig_argv = NULL;
 static int orig_argc = 0;
@@ -1963,13 +1971,15 @@
 pymain_read_conf(_PyMain *pymain, _Py_CommandLineDetails *cmdline)
 {
     int res = -1;
-
+#if NO_LOCALECONV
+    char *oldloc = NULL;
+#else
     char *oldloc = _PyMem_RawStrdup(setlocale(LC_ALL, NULL));
     if (oldloc == NULL) {
         pymain->err = _Py_INIT_NO_MEMORY();
         goto done;
     }
-
+#endif
     /* Reconfigure the locale to the default for this process */
     _Py_SetLocaleFromEnv(LC_ALL);
 
