--- cpython-master/Modules/signalmodule.c	2018-01-23 22:26:07.000000000 +0100
+++ cpython-master.api19/Modules/signalmodule.c	2018-01-23 23:55:20.000000000 +0100
@@ -40,6 +40,13 @@
 #define SIG_ERR ((PyOS_sighandler_t)(-1))
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    #define SIGRTMIN 32
+    #define SIGRTMAX _NSIG
+#endif
+//PMPP API<21
+
 #ifndef NSIG
 # if defined(_NSIG)
 #  define NSIG _NSIG            /* For BSD/SysV */
