--- cpython-3.8.5/Modules/_hashopenssl.c	2020-07-20 15:01:32.000000000 +0200
+++ python3-wasm/Modules/_hashopenssl.c	2020-08-30 17:22:21.654931539 +0200
@@ -27,8 +27,11 @@
 #include "openssl/err.h"
 
 #ifndef OPENSSL_THREADS
+#if __EMSCRIPTEN__
+#else
 #  error "OPENSSL_THREADS is not defined, Python requires thread-safe OpenSSL"
 #endif
+#endif
 
 #if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
 /* OpenSSL < 1.1.0 */
