--- cpython-3.8.3/Modules/clinic/signalmodule.c.h	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-wasm/Modules/clinic/signalmodule.c.h	2020-06-07 12:53:04.591798747 +0200
@@ -396,9 +396,13 @@
     if (how == -1 && PyErr_Occurred()) {
         goto exit;
     }
+	#if __EMSCRIPTEN__
+		#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+	#else
     if (!_Py_Sigset_Converter(args[1], &mask)) {
         goto exit;
     }
+	#endif
     return_value = signal_pthread_sigmask_impl(module, how, mask);
 
 exit:
