--- cpython-3.8.5/Parser/myreadline.c	2020-07-20 15:01:32.000000000 +0200
+++ python3-wasm/Parser/myreadline.c	2020-08-28 12:53:03.848088103 +0200
@@ -11,10 +11,6 @@
 
 #include "Python.h"
 #include "pycore_pystate.h"
-#ifdef MS_WINDOWS
-#define WIN32_LEAN_AND_MEAN
-#include "windows.h"
-#endif /* MS_WINDOWS */
 
 
 PyThreadState* _PyOS_ReadlineTState = NULL;
@@ -315,6 +311,11 @@
         return NULL;
     case -1: /* EOF */
     case -2: /* Error */
+    case -3:
+        #ifdef __EMSCRIPTEN__
+        #warning ------------- no no no i won't go there --------------
+        #endif
+        break;
     default: /* Shouldn't happen */
         *p = '\0';
         break;
