--- cpython-3.8.5/Python/pylifecycle.c	2020-07-20 15:01:32.000000000 +0200
+++ python3-wasm/Python/pylifecycle.c	2020-08-28 12:53:03.856087998 +0200
@@ -2369,6 +2369,8 @@
     if (sigaction(sig, NULL, &context) == -1)
         return SIG_ERR;
     return context.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t handler;
 /* Special signal handling for the secure CRT in Visual Studio 2005 */
@@ -2415,6 +2417,8 @@
     if (sigaction(sig, &context, &ocontext) == -1)
         return SIG_ERR;
     return ocontext.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t oldhandler;
     oldhandler = signal(sig, handler);
