--- cpython-3.8.5/Python/fileutils.c	2020-07-20 15:01:32.000000000 +0200
+++ python3-wasm/Python/fileutils.c	2020-08-28 12:53:03.851088063 +0200
@@ -1091,6 +1091,9 @@
 static int
 set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
 {
+#ifdef __EMSCRIPTEN__
+    return 0;
+#else
 #ifdef MS_WINDOWS
     HANDLE handle;
     DWORD flags;
@@ -1214,6 +1217,7 @@
     }
     return 0;
 #endif
+#endif
 }
 
 /* Make the file descriptor non-inheritable.
