--- python3/Lib/locale.py	2019-10-15 00:32:36.000000000 +0200
+++ python3-android/Lib/locale.py	2019-10-23 18:14:07.991613058 +0200
@@ -46,6 +46,10 @@
 
     from _locale import *
 
+    if hasattr(sys, 'getandroidapilevel'):
+        if sys.getandroidapilevel()<21:
+            raise ImportError("android-api19")
+
 except ImportError:
 
     # Locale emulation
@@ -88,6 +92,9 @@
         """ setlocale(integer,string=None) -> string.
             Activates/queries locale processing.
         """
+        if category>LC_ALL:
+            raise Error('_locale emulation invalid category')
+
         if value not in (None, '', 'C'):
             raise Error('_locale emulation only supports "C" locale')
         return 'C'
