--- panda3d-input-overhaul/makepanda/makepandacore.py	2018-01-28 14:44:54.000000000 +0100
+++ panda3d-input-overhaul.api19/makepanda/makepandacore.py	2018-01-28 19:50:59.000000000 +0100
@@ -442,12 +442,16 @@
 
 def GetCC():
     if TARGET in ('darwin', 'freebsd', 'android'):
+        if os.getenv('CLANG'):
+            return os.getenv('CLANG',None)
         return os.environ.get('CC', TOOLCHAIN_PREFIX + 'clang')
     else:
         return os.environ.get('CC', TOOLCHAIN_PREFIX + 'gcc')
 
 def GetCXX():
     if TARGET in ('darwin', 'freebsd', 'android'):
+        if os.getenv('CLANG'):
+            return '%s++' % os.getenv('CLANG',None)
         return os.environ.get('CXX', TOOLCHAIN_PREFIX + 'clang++')
     else:
         return os.environ.get('CXX', TOOLCHAIN_PREFIX + 'g++')
@@ -2034,10 +2038,10 @@
         if GetTarget() == 'darwin':
             py_libs = glob.glob(tp_python + "/lib/libpython[0-9].[0-9].dylib")
         else:
-            py_libs = glob.glob(tp_python + "/lib/libpython[0-9].[0-9].so")
+            py_libs = glob.glob(tp_python + "/lib/libpython[0-9].[0-9]%sm.so" % '')
 
         if len(py_libs) == 0:
-            py_libs = glob.glob(tp_python + "/lib/libpython[0-9].[0-9].a")
+            py_libs = glob.glob(tp_python + "/lib/libpython[0-9].[0-9]%sm.a" % '')
 
         if len(py_libs) == 0:
             exit("Could not find the Python library in %s." % (tp_python))
@@ -2045,7 +2049,8 @@
             exit("Found multiple Python libraries in %s." % (tp_python))
 
         py_lib = os.path.basename(py_libs[0])
-        SDK["PYTHONVERSION"] = "python" + py_lib[9] + "." + py_lib[11]
+        #"python" + py_lib[9] + "." + py_lib[11]
+        SDK["PYTHONVERSION"] = py_lib.replace('lib','').rsplit('.',1)[0]
         SDK["PYTHONEXEC"] = tp_python + "/bin/" + SDK["PYTHONVERSION"]
         SDK["PYTHON"] = tp_python + "/include/" + SDK["PYTHONVERSION"]
 
@@ -2388,7 +2393,7 @@
     SDK["ANDROID_NDK"] = ndk_root
 
     # Determine the toolchain location.
-    gcc_ver = '4.8'
+    gcc_ver = '4.9'
     arch = GetTargetArch()
     if arch == 'armv7a' or arch == 'arm':
         arch = 'arm'
