--- panda3d-input-overhaul/makepanda/makepanda.py	2018-01-28 14:44:54.000000000 +0100
+++ panda3d-input-overhaul.api19/makepanda/makepanda.py	2018-01-31 09:15:25.533772145 +0100
@@ -1745,7 +1745,7 @@
             cmd += " -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now"
             if GetTargetArch() == 'armv7a':
                 cmd += " -march=armv7-a -Wl,--fix-cortex-a8"
-            cmd += ' -lc -lm'
+            cmd += ' -lc -lm -l%s' % os.getenv('PANDA_PYLIB','python3.7m')
         else:
             cmd += " -pthread"
 
