--- panda3d-input-overhaul/panda/src/android/PandaActivity.java	2018-02-04 11:37:09.000000000 +0100
+++ panda3d-input-overhaul.api19/panda/src/android/PandaActivity.java	2018-02-19 02:32:16.803446786 +0100
@@ -12,16 +12,17 @@
  */
 
 package org.panda3d.android;
-
+/*
 import android.app.NativeActivity;
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
 import org.panda3d.android.NativeIStream;
-
+*/
 /**
  * The entry point for a Panda-based activity.  Loads the Panda libraries and
  * also provides some utility functions.
  */
+/*
 public class PandaActivity extends NativeActivity {
     protected static BitmapFactory.Options readBitmapSize(long istreamPtr) {
         BitmapFactory.Options options = new BitmapFactory.Options();
@@ -53,3 +54,4 @@
         System.loadLibrary("pandagles");
     }
 }
+*/
