--- panda3d-input-overhaul/panda/src/device/p3device_composite1.cxx	2018-01-28 14:44:54.000000000 +0100
+++ panda3d-input-overhaul.api19/panda/src/device/p3device_composite1.cxx	2018-02-02 11:46:32.613559543 +0100
@@ -1,3 +1,17 @@
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 20
+    #include <linux/input.h>
+    #define EV_CNT (EV_MAX+1)
+    #define KEY_CNT (KEY_MAX+1)
+    #define ABS_CNT (ABS_MAX+1)
+    #define FF_CNT (FF_MAX+1)
+    #define BTN_TRIGGER_HAPPY1 0x2c0
+    #define BTN_TRIGGER_HAPPY2 0x2c1
+    #define BTN_TRIGGER_HAPPY3 0x2c2
+    #define BTN_TRIGGER_HAPPY4 0x2c3
+
+    #define IN_CLOEXEC O_CLOEXEC
+    #define IN_NONBLOCK O_NONBLOCK
+#endif
 
 #include "clientAnalogDevice.cxx"
 #include "clientBase.cxx"
