--- panda3d-input-overhaul/panda/src/device/linuxInputDeviceManager.cxx	2018-01-28 14:44:54.000000000 +0100
+++ panda3d-input-overhaul.api19/panda/src/device/linuxInputDeviceManager.cxx	2018-02-02 11:46:32.587559543 +0100
@@ -32,7 +32,12 @@
 LinuxInputDeviceManager::
 LinuxInputDeviceManager() {
   // Use inotify to watch /dev/input for hotplugging of devices.
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 20
+  _inotify_fd = inotify_init();
+  fcntl( _inotify_fd , IN_NONBLOCK | IN_CLOEXEC );
+#else
   _inotify_fd = inotify_init1(IN_NONBLOCK | IN_CLOEXEC);
+#endif
 
   if (_inotify_fd < 0) {
     device_cat.error()
