--- Python-3.7.12/Lib/runpy.py	2021-09-04 05:49:21.000000000 +0200
+++ Python-3.7.12-pydk/Lib/runpy.py	2022-03-16 12:55:45.929067320 +0100
@@ -133,7 +133,7 @@
         msg = "Error while finding module specification for {!r} ({}: {})"
         raise error(msg.format(mod_name, type(ex).__name__, ex)) from ex
     if spec is None:
-        raise error("No module named %s" % mod_name)
+        raise error("No module named5 %s" % mod_name)
     if spec.submodule_search_locations is not None:
         if mod_name == "__main__" or mod_name.endswith(".__main__"):
             raise error("Cannot use package as __main__ module")
