--- python3/Modules/mmapmodule.c	2020-02-24 22:36:25.000000000 +0100
+++ python3-android/Modules/mmapmodule.c	2020-03-29 18:49:07.039978523 +0200
@@ -79,6 +79,12 @@
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
+
 typedef enum
 {
     ACCESS_DEFAULT,
