--- python3/Modules/_decimal/libmpdec/io.c	2020-02-24 22:36:25.000000000 +0100
+++ python3-android/Modules/_decimal/libmpdec/io.c	2020-03-29 18:49:07.035978595 +0200
@@ -871,9 +871,15 @@
         if (*spec->sep) {
             return 0;
         }
+
         spec->type = *cp++;
         spec->type = (spec->type == 'N') ? 'G' : 'g';
-        lc = localeconv();
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+        #warning "====================================== broken_localeconv ================================"
+        lc = broken_localeconv();
+#else
+        lc = localeconv(); //PATCHED
+#endif
         spec->dot = lc->decimal_point;
         spec->sep = lc->thousands_sep;
         spec->grouping = lc->grouping;
