--- python3/Python/formatter_unicode.c	2020-02-24 22:36:25.000000000 +0100
+++ python3-android/Python/formatter_unicode.c	2020-03-29 18:49:07.051978305 +0200
@@ -690,6 +690,7 @@
 
 static const char no_grouping[1] = {CHAR_MAX};
 
+
 /* Find the decimal point character(s?), thousands_separator(s?), and
    grouping description, either for the current locale if type is
    LT_CURRENT_LOCALE, a hard-coded locale if LT_DEFAULT_LOCALE or
