--- Python-3.7.12/Python/import.c	2021-09-04 05:49:21.000000000 +0200
+++ Python-3.7.12-pydk/Python/import.c	2022-03-16 12:55:46.741040430 +0100
@@ -2151,7 +2151,9 @@
         Py_DECREF(name);
         return NULL;
     }
-
+#if __ANDROID__
+    printf("2151: name: %s, path:%s\n",PyUnicode_AsUTF8(name),PyUnicode_AsUTF8(path));
+#endif
     mod = _PyImport_FindExtensionObject(name, path);
     if (mod != NULL || PyErr_Occurred()) {
         Py_DECREF(name);
