--- openssl-1.1.1d/Configurations/10-main.conf	2019-09-10 15:13:07.000000000 +0200
+++ openssl-1.1.1d.pydk/Configurations/10-main.conf	2019-12-16 18:48:37.644606094 +0100
@@ -642,10 +642,10 @@
         CXXFLAGS         => picker(default => "-Wall",
                                    debug   => "-O0 -g",
                                    release => "-O3"),
-        cflags           => threads("-pthread"),
-        cxxflags         => combine("-std=c++11", threads("-pthread")),
+        cflags           => "",
+        cxxflags         => combine("-std=c++11"),
         lib_cppflags     => "-DOPENSSL_USE_NODELETE",
-        ex_libs          => add("-ldl", threads("-pthread")),
+        ex_libs          => add("-ldl"),
         bn_ops           => "BN_LLONG RC4_CHAR",
         thread_scheme    => "pthreads",
         dso_scheme       => "dlfcn",
