--- Python-3.7.10/configure.ac	2021-02-16 02:29:22.000000000 +0100
+++ python3-wasm/configure.ac	2021-05-04 18:24:59.658254732 +0200
@@ -376,6 +376,13 @@
 	*-*-linux*)
 		ac_sys_system=Linux
 		;;
+    wasm32-unknown-wasi)
+        ac_sys_system=Linux-wasi32
+		_host_cpu=
+		;;
+	wasm-*-*)
+		_host_cpu=
+		;;
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
@@ -423,6 +430,12 @@
 	*-*-cygwin*)
 		_host_cpu=
 		;;
+	wasm-*-*)
+		ac_sys_system=Emscripten
+		;;
+	wasm32-unknown-wasi)
+		ac_sys_system=wasi32
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -1141,6 +1154,16 @@
 	      PY3LIBRARY=libpython3.so
 	  fi
           ;;
+    Emscripten)
+	  LDLIBRARY='libpython$(LDVERSION).bc'
+	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
+	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
+	  INSTSONAME="$LDLIBRARY"
+	  if test "$with_pydebug" != yes
+          then
+	      PY3LIBRARY=libpython3.bc
+	  fi
+	  ;;
     Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
 	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
@@ -2513,8 +2536,11 @@
 # SHLIB_SUFFIX is the extension of shared libraries `(including the dot!)
 # -- usually .so, .sl on HP-UX, .dll on Cygwin
 AC_MSG_CHECKING(the extension of shared libraries)
+
 if test -z "$SHLIB_SUFFIX"; then
 	case $ac_sys_system in
+	Emscripten)
+	    SHLIB_SUFFIX=.js;;
 	hp*|HP*)
 		case `uname -m` in
 			ia64) SHLIB_SUFFIX=.so;;
@@ -2525,6 +2551,12 @@
 	*)	   SHLIB_SUFFIX=.so;;
 	esac
 fi
+
+case $ac_sys_system in
+Emscripten)
+    SHLIB_SUFFIX=.js;;
+esac
+
 AC_MSG_RESULT($SHLIB_SUFFIX)
 
 # LDSHARED is the ld *command* used to create shared library
@@ -4764,6 +4796,10 @@
 case $ac_sys_system in
     Linux*|GNU*|Darwin)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+    Emscripten)
+        SOABI="cpython-${VERSION}-emscripten"
+        EXT_SUFFIX=".extsu"
+        ;;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac
