--- Python-3.7.10/Lib/subprocess.py	2021-02-16 02:29:22.000000000 +0100
+++ python3-wasm/Lib/subprocess.py	2021-05-04 18:24:59.670254608 +0200
@@ -149,7 +149,10 @@
                                lpAttributeList=attr_list)
 
 else:
-    import _posixsubprocess
+    try:
+        import _posixsubprocess
+    except:
+        print(__file__,"no _posixsubprocess",file=sys.stderr)
     import select
     import selectors
     import threading
