--- Python-3.7.10/Python/fileutils.c	2021-02-16 02:29:22.000000000 +0100
+++ python3-wasm/Python/fileutils.c	2021-05-04 18:24:59.706254238 +0200
@@ -994,6 +994,9 @@
 static int
 set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
 {
+#ifdef __EMSCRIPTEN__
+    return 0;
+#else
 #ifdef MS_WINDOWS
     HANDLE handle;
     DWORD flags;
@@ -1117,6 +1120,7 @@
     }
     return 0;
 #endif
+#endif
 }
 
 /* Make the file descriptor non-inheritable.
