--- cpython-3.7.1rc1/Modules/_localemodule.c	2018-09-26 08:16:09.000000000 +0200
+++ cpython-3.7.1rc1.api19/Modules/_localemodule.c	2018-09-29 01:29:03.641351545 +0200
@@ -103,6 +103,10 @@
         PyErr_SetString(Error, "invalid locale category");
         return NULL;
     }
+#else
+#ifdef __ANDROID__
+    return PyUnicode_FromFormat("%s", "C");
+#endif
 #endif
 
     if (locale) {
@@ -144,7 +148,16 @@
     }
 
     /* if LC_NUMERIC is different in the C library, use saved value */
-    l = localeconv();
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    /* Don't even try on Android's broken locale.h. */
+    /* maybe can use c++ stdlib to implement */
+    goto failed;
+#else
+     /* if LC_NUMERIC is different in the C library, use saved value */
+     l = localeconv(); //PATCHED
+#endif
+//PMPP API<21
 
     /* hopefully, the localeconv result survives the C library calls
        involved herein */
