--- cpython-3.7.1rc1/Modules/signalmodule.c	2018-09-26 08:16:09.000000000 +0200
+++ cpython-3.7.1rc1.api19/Modules/signalmodule.c	2018-09-29 01:29:03.705351544 +0200
@@ -40,6 +40,13 @@
 #define SIG_ERR ((PyOS_sighandler_t)(-1))
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    #define SIGRTMIN 32
+    #define SIGRTMAX _NSIG
+#endif
+//PMPP API<21
+
 #ifndef NSIG
 # if defined(_NSIG)
 #  define NSIG _NSIG            /* For BSD/SysV */
