--- cpython-3.7.1rc1/Modules/mmapmodule.c	2018-09-26 08:16:09.000000000 +0200
+++ cpython-3.7.1rc1.api19/Modules/mmapmodule.c	2018-09-29 01:29:03.667351545 +0200
@@ -79,6 +79,12 @@
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
+
 typedef enum
 {
     ACCESS_DEFAULT,
