--- Python-3.8.0b3/configure.ac	2019-07-29 15:26:01.000000000 +0200
+++ Python-3.8.0b3.em/configure.ac	2019-08-29 10:21:44.740154515 +0200
@@ -379,6 +379,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	asmjs-*-*)
+		_host_cpu=
+		;;
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
@@ -431,6 +434,9 @@
 	*-*-vxworks*)
 		_host_cpu=$host_cpu
 		;;
+	asmjs-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -1119,6 +1125,16 @@
 	      PY3LIBRARY=libpython3.so
 	  fi
           ;;
+    Emscripten)
+	  LDLIBRARY='libpython$(LDVERSION).bc'
+	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
+	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
+	  INSTSONAME="$LDLIBRARY"
+	  if test "$with_pydebug" != yes
+          then
+	      PY3LIBRARY=libpython3.bc
+	  fi
+	  ;;
     Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
 	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
@@ -2490,8 +2506,11 @@
 # SHLIB_SUFFIX is the extension of shared libraries `(including the dot!)
 # -- usually .so, .sl on HP-UX, .dll on Cygwin
 AC_MSG_CHECKING(the extension of shared libraries)
+
 if test -z "$SHLIB_SUFFIX"; then
 	case $ac_sys_system in
+	Emscripten)
+	    SHLIB_SUFFIX=.js;;
 	hp*|HP*)
 		case `uname -m` in
 			ia64) SHLIB_SUFFIX=.so;;
@@ -2502,6 +2521,12 @@
 	*)	   SHLIB_SUFFIX=.so;;
 	esac
 fi
+
+case $ac_sys_system in
+Emscripten)
+    SHLIB_SUFFIX=.js;;
+esac
+
 AC_MSG_RESULT($SHLIB_SUFFIX)
 
 # LDSHARED is the ld *command* used to create shared library
@@ -4644,6 +4669,10 @@
 case $ac_sys_system in
     Linux*|GNU*|Darwin|VxWorks)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+    Emscripten)
+        SOABI="cpython-${VERSION}-emscripten"
+        EXT_SUFFIX=".extsu"
+        ;;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac
