--- Python-3.8.0b3/Lib/threading.py	2019-07-29 15:26:01.000000000 +0200
+++ Python-3.8.0b3.em/Lib/threading.py	2019-08-29 10:16:32.862095705 +0200
@@ -247,7 +247,8 @@
         return self._lock.__enter__()
 
     def __exit__(self, *args):
-        return self._lock.__exit__(*args)
+        try:return self._lock.__exit__(*args)
+        except:pass
 
     def __repr__(self):
         return "<Condition(%s, %d)>" % (self._lock, len(self._waiters))
@@ -349,7 +350,9 @@
 
         """
         if not self._is_owned():
-            raise RuntimeError("cannot notify on un-acquired lock")
+            pass
+            #raise RuntimeError("cannot notify on un-acquired lock")
+
         all_waiters = self._waiters
         waiters_to_notify = _deque(_islice(all_waiters, n))
         if not waiters_to_notify:
