--- Python-3.8.0b3/Python/import.c	2019-07-29 15:26:01.000000000 +0200
+++ Python-3.8.0b3.em/Python/import.c	2019-08-29 10:16:32.906094299 +0200
@@ -2025,6 +2025,20 @@
     list = PyList_New(0);
     if (list == NULL)
         return NULL;
+
+#ifdef __EMSCRIPTEN__
+        PyObject *item = PyUnicode_FromString( ".cpython-38-em.wasm");
+        if (item == NULL) {
+            Py_DECREF(list);
+            return NULL;
+        }
+        if (PyList_Append(list, item) < 0) {
+            Py_DECREF(list);
+            Py_DECREF(item);
+            return NULL;
+        }
+        Py_DECREF(item);
+#else
 #ifdef HAVE_DYNAMIC_LOADING
     const char *suffix;
     unsigned int index = 0;
@@ -2044,6 +2058,7 @@
         index += 1;
     }
 #endif
+#endif
     return list;
 }
 
