--- cpython-3.8.0a1/Include/pyport.h	2019-02-03 14:17:38.000000000 +0100
+++ cpython-3.8.0a1.wasm/Include/pyport.h	2019-02-04 20:22:02.106068740 +0100
@@ -698,7 +698,11 @@
  * rather than waiting for integer multiplication to trigger bogus
  * overflows.
  */
+#ifndef __EMSCRIPTEN__
 #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
+#else
+#warning "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
+#endif
 #endif
 
 #ifdef __cplusplus
