--- cpython-3.8.0a1/Modules/clinic/signalmodule.c.h	2019-02-03 14:17:38.000000000 +0100
+++ cpython-3.8.0a1.wasm/Modules/clinic/signalmodule.c.h	2019-02-04 21:09:20.422893693 +0100
@@ -396,9 +396,13 @@
     if (how == -1 && PyErr_Occurred()) {
         goto exit;
     }
+	#if __EMSCRIPTEN__
+		#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+	#else
     if (!_Py_Sigset_Converter(args[1], &mask)) {
         goto exit;
     }
+	#endif
     return_value = signal_pthread_sigmask_impl(module, how, mask);
 
 exit:
