--- cpython-3.8.0a1/Modules/readline.c	2019-02-03 14:17:38.000000000 +0100
+++ cpython-3.8.0a1.wasm/Modules/readline.c	2019-02-04 20:22:02.126067670 +0100
@@ -1184,6 +1184,11 @@
 
     completed_input_string = not_done_reading;
 
+    #ifdef __EMSCRIPTEN__
+        #error no no no i won't go there
+    #endif
+
+
     while (completed_input_string == not_done_reading) {
         int has_input = 0, err = 0;
 
