--- cpython-3.8.0a1/Modules/posixmodule.c	2019-02-03 14:17:38.000000000 +0100
+++ cpython-3.8.0a1.wasm/Modules/posixmodule.c	2019-02-04 21:05:32.688285994 +0100
@@ -5211,9 +5211,13 @@
 
    if (setsigmask) {
         sigset_t set;
+		#if __EMSCRIPTEN__
+			#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+		#else
         if (!_Py_Sigset_Converter(setsigmask, &set)) {
             goto fail;
         }
+		#endif
         errno = posix_spawnattr_setsigmask(attrp, &set);
         if (errno) {
             posix_error();
@@ -5224,9 +5228,13 @@
 
     if (setsigdef) {
         sigset_t set;
+		#if __EMSCRIPTEN__
+			#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+		#else
         if (!_Py_Sigset_Converter(setsigdef, &set)) {
             goto fail;
         }
+		#endif
         errno = posix_spawnattr_setsigdefault(attrp, &set);
         if (errno) {
             posix_error();
