--- cpython-3.8.0a1/Python/import.c	2019-02-03 14:17:38.000000000 +0100
+++ cpython-3.8.0a1.wasm/Python/import.c	2019-02-04 20:22:02.150066385 +0100
@@ -1974,6 +1974,20 @@
     list = PyList_New(0);
     if (list == NULL)
         return NULL;
+
+#ifdef __EMSCRIPTEN__
+        PyObject *item = PyUnicode_FromString( ".cpython-38-em.wasm");
+        if (item == NULL) {
+            Py_DECREF(list);
+            return NULL;
+        }
+        if (PyList_Append(list, item) < 0) {
+            Py_DECREF(list);
+            Py_DECREF(item);
+            return NULL;
+        }
+        Py_DECREF(item);
+#else
 #ifdef HAVE_DYNAMIC_LOADING
     while ((suffix = _PyImport_DynLoadFiletab[index])) {
         PyObject *item = PyUnicode_FromString(suffix);
@@ -1990,6 +2004,7 @@
         index += 1;
     }
 #endif
+#endif
     return list;
 }
 
