--- Python-3.11.0a4/Modules/readline.c	2022-01-13 20:38:15.000000000 +0100
+++ Python-3.11.0a4.patch/Modules/readline.c	2022-01-20 08:40:17.654275099 +0100
@@ -1326,7 +1326,9 @@
         int has_input = 0, err = 0;
 
         while (!has_input)
-        {               struct timeval timeout = {0, 100000}; /* 0.1 seconds */
+        {
+
+            struct timeval timeout = {0, 10000}; /* 0.01 seconds */
 
             /* [Bug #1552726] Only limit the pause if an input hook has been
                defined.  */
