--- Python-3.11.0a4/Parser/tokenizer.c	2022-01-13 20:38:15.000000000 +0100
+++ Python-3.11.0a4.patch/Parser/tokenizer.c	2022-01-21 18:36:30.284857129 +0100
@@ -855,7 +855,7 @@
         tok->done = E_INTERACT_STOP;
         return 1;
     }
-    char *newtok = PyOS_Readline(stdin, stdout, tok->prompt);
+    char *newtok = PyOS_Readline(tok->fp ? tok->fp : stdin, stdout, tok->prompt);
     if (newtok != NULL) {
         char *translated = translate_newlines(newtok, 0, tok);
         PyMem_Free(newtok);
