--- Python-3.7.10/Lib/platform.py	2021-02-16 02:29:22.000000000 +0100
+++ Python-3.7.10-aosp/Lib/platform.py	2021-03-04 15:06:27.768071521 +0100
@@ -188,6 +188,9 @@
         The file is read and scanned in chunks of chunksize bytes.
 
     """
+    if hasattr(sys, 'getandroidapilevel'):
+        return 'libc', f'{sys.getandroidapilevel()}-bionic'
+
     V = _comparable_version
     if hasattr(os.path, 'realpath'):
         # Python 2.2 introduced os.path.realpath(); it is used
