--- Python-3.7.10/Lib/locale.py	2021-02-16 02:29:22.000000000 +0100
+++ Python-3.7.10-aosp/Lib/locale.py	2021-03-04 15:06:27.760071792 +0100
@@ -46,6 +46,10 @@
 
     from _locale import *
 
+    if hasattr(sys, 'getandroidapilevel'):
+        if sys.getandroidapilevel()<21:
+            raise ImportError("android-api19")
+
 except ImportError:
 
     # Locale emulation
@@ -88,6 +92,9 @@
         """ setlocale(integer,string=None) -> string.
             Activates/queries locale processing.
         """
+        if category>LC_ALL:
+            raise Error('_locale emulation invalid category')
+
         if value not in (None, '', 'C'):
             raise Error('_locale emulation only supports "C" locale')
         return 'C'
