--- Python-3.7.10/Modules/signalmodule.c	2021-02-16 02:29:22.000000000 +0100
+++ Python-3.7.10-aosp/Modules/signalmodule.c	2021-03-04 15:06:27.808070166 +0100
@@ -40,6 +40,13 @@
 #define SIG_ERR ((PyOS_sighandler_t)(-1))
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    #define SIGRTMIN 32
+    #define SIGRTMAX _NSIG
+#endif
+//PMPP API<21
+
 #ifndef NSIG
 # if defined(_NSIG)
 #  define NSIG _NSIG            /* For BSD/SysV */
