--- Python-3.7.10/Python/import.c	2021-02-16 02:29:22.000000000 +0100
+++ Python-3.7.10-aosp/Python/import.c	2021-03-04 15:06:27.832069351 +0100
@@ -2151,7 +2151,9 @@
         Py_DECREF(name);
         return NULL;
     }
-
+#if __ANDROID__
+    printf("2151: name: %s, path:%s\n",PyUnicode_AsUTF8(name),PyUnicode_AsUTF8(path));
+#endif
     mod = _PyImport_FindExtensionObject(name, path);
     if (mod != NULL || PyErr_Occurred()) {
         Py_DECREF(name);
