--- Python-3.7.10/Python/pystrtod.c	2021-02-16 02:29:22.000000000 +0100
+++ Python-3.7.10-aosp/Python/pystrtod.c	2021-03-04 15:06:27.840069079 +0100
@@ -177,8 +177,14 @@
 
     fail_pos = NULL;
 
-    locale_data = localeconv();
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+    locale_data = broken_localeconv();
     decimal_point = locale_data->decimal_point;
+#else
+    locale_data = localeconv(); //PATCHED
+    decimal_point = locale_data->decimal_point;
+#endif // __ANDROID_API__
+
     decimal_point_len = strlen(decimal_point);
 
     assert(decimal_point_len != 0);
