--- Python-3.7.10/Python/importdl.c	2021-02-16 02:29:22.000000000 +0100
+++ Python-3.7.10-aosp/Python/importdl.c	2021-03-04 15:06:27.836069216 +0100
@@ -126,6 +126,9 @@
     pathbytes = PyUnicode_EncodeFSDefault(path);
     if (pathbytes == NULL)
         goto error;
+#if __ANDROID__
+#warning "FIXME: supress lib from "PyInit_lib"
+#endif
     exportfunc = _PyImport_FindSharedFuncptr(hook_prefix, name_buf,
                                              PyBytes_AS_STRING(pathbytes),
                                              fp);
