--- Python-3.7.0b5/configure	2018-05-31 01:50:49.000000000 +0200
+++ Python-3.7.0b5.em/configure	2018-06-01 09:30:09.529817813 +0200
@@ -3257,6 +3257,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	asmjs-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3301,6 +3304,9 @@
 	*-*-cygwin*)
 		_host_cpu=
 		;;
+	asmjs-*-*)
+		_host_cpu=
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -11506,6 +11512,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -11531,6 +11540,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -12180,7 +12192,9 @@
 #ifdef HAVE_GRP_H
 #include <grp.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error setgroups
+#endif
 int
 main ()
 {
@@ -12718,7 +12732,9 @@
 #else
 #include <sys/types.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error makedev
+#endif
 int
 main ()
 {
