--- Python-3.7.0b5/Lib/subprocess.py	2018-05-31 01:50:49.000000000 +0200
+++ Python-3.7.0b5.em/Lib/subprocess.py	2018-06-01 09:30:09.564817812 +0200
@@ -40,10 +40,9 @@
 getstatusoutput(...): Runs a command in the shell, waits for it to complete,
     then returns a (exitcode, output) tuple
 """
-
 import sys
 _mswindows = (sys.platform == "win32")
-
+_emscripten = (sys.platform == "emscripten")
 import io
 import os
 import time
@@ -136,7 +135,8 @@
             self.wShowWindow = wShowWindow
             self.lpAttributeList = lpAttributeList or {"handle_list": []}
 else:
-    import _posixsubprocess
+    if not _emscripten:
+        import _posixsubprocess
     import select
     import selectors
     import threading
