--- Python-3.7.0b5/Python/pylifecycle.c	2018-05-31 01:50:49.000000000 +0200
+++ Python-3.7.0b5.em/Python/pylifecycle.c	2018-06-01 09:30:09.673817810 +0200
@@ -2202,6 +2202,8 @@
     if (sigaction(sig, NULL, &context) == -1)
         return SIG_ERR;
     return context.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t handler;
 /* Special signal handling for the secure CRT in Visual Studio 2005 */
@@ -2248,6 +2250,8 @@
     if (sigaction(sig, &context, &ocontext) == -1)
         return SIG_ERR;
     return ocontext.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t oldhandler;
     oldhandler = signal(sig, handler);
