--- webgl-port.git/cmake/macros/Python.cmake	2022-01-25 14:21:19.028785884 +0100
+++ mobile-sandbox/cmake/macros/Python.cmake	2022-02-09 21:41:17.907574181 +0100
@@ -48,7 +48,12 @@
   string(REPLACE "." "/" slash_namespace "${namespace}")
 
   add_library(${target} ${MODULE_TYPE} ${sources})
-  target_link_libraries(${target} PKG::PYTHON)
+
+  if (ANDROID)
+    target_link_libraries(${target} "python${PYMAJOR}.${PYMINOR}" "ffi" "z"  "bz2" "lzma" "brokenthings")
+  else()
+    target_link_libraries(${target} PKG::PYTHON)
+  endif()
 
   if(BUILD_SHARED_LIBS)
     set(_outdir "${PANDA_OUTPUT_DIR}/${slash_namespace}")
