--- webgl-port.git/panda/src/viewdisplay/config_viewdisplay.h	2022-02-07 10:52:42.697876520 +0100
+++ mobile-sandbox/panda/src/viewdisplay/config_viewdisplay.h	2022-02-09 21:41:17.912574140 +0100
@@ -0,0 +1,92 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file config_viewdisplay.h
+ * @author pmp-p
+ * @date 2020-04-07
+ */
+
+#ifndef CONFIG_VIEWDISPLAY_H
+#define CONFIG_VIEWDISPLAY_H
+
+#include "pandabase.h"
+#include "filename.h"
+#include "notifyCategoryProxy.h"
+#include "configVariableString.h"
+#include "configVariableBool.h"
+#include "configVariableInt.h"
+
+
+#if defined(OPENGLES_1) && defined(OPENGLES_2)
+  #error OPENGLES_1 and OPENGLES_2 cannot be defined at the same time!
+#endif
+#if !defined(OPENGLES_1) && !defined(OPENGLES_2)
+  #error Either OPENGLES_1 or OPENGLES_2 must be defined when compiling viewdisplay!
+#endif
+
+#ifdef OPENGLES_2
+  NotifyCategoryDecl(viewdisplay, EXPCL_PANDAGLES2, EXPTP_PANDAGLES2);
+
+  extern EXPCL_PANDAGLES2 void init_libviewdisplay();
+  extern EXPCL_PANDAGLES2 const std::string get_egl_error_string(int error);
+#else
+    #error GLES1 dropped
+  NotifyCategoryDecl(viewdisplay, EXPCL_PANDAGLES, EXPTP_PANDAGLES);
+
+  extern EXPCL_PANDAGLES void init_libviewdisplay();
+  extern EXPCL_PANDAGLES const std::string get_egl_error_string(int error);
+#endif
+
+
+#endif
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file config_viewdisplay.h
+ * @author pmp-p
+ * @date 2020-04-07
+ */
+
+#ifndef CONFIG_VIEWDISPLAY_H
+#define CONFIG_VIEWDISPLAY_H
+
+#include "pandabase.h"
+#include "filename.h"
+#include "notifyCategoryProxy.h"
+#include "configVariableString.h"
+#include "configVariableBool.h"
+#include "configVariableInt.h"
+
+
+#if defined(OPENGLES_1) && defined(OPENGLES_2)
+  #error OPENGLES_1 and OPENGLES_2 cannot be defined at the same time!
+#endif
+#if !defined(OPENGLES_1) && !defined(OPENGLES_2)
+  #error Either OPENGLES_1 or OPENGLES_2 must be defined when compiling viewdisplay!
+#endif
+
+#ifdef OPENGLES_2
+  NotifyCategoryDecl(viewdisplay, EXPCL_PANDAGLES2, EXPTP_PANDAGLES2);
+
+  extern EXPCL_PANDAGLES2 void init_libviewdisplay();
+  extern EXPCL_PANDAGLES2 const std::string get_egl_error_string(int error);
+#else
+    #error GLES1 dropped
+  NotifyCategoryDecl(viewdisplay, EXPCL_PANDAGLES, EXPTP_PANDAGLES);
+
+  extern EXPCL_PANDAGLES void init_libviewdisplay();
+  extern EXPCL_PANDAGLES const std::string get_egl_error_string(int error);
+#endif
+
+
+#endif
