--- panda3d-980c6bb38961c13e0890990651d05df3550cf30a/panda/src/viewdisplay/viewGraphicsWindow.I	2020-04-13 05:43:16.709728110 +0200
+++ panda3d-android/panda/src/viewdisplay/viewGraphicsWindow.I	2020-04-13 03:26:24.900320771 +0200
@@ -0,0 +1,12 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file viewGraphicsWindow.I
+ * @author pmp-p
+ * @date 2020-04-07
+ */
