--- webgl-port.git/panda/src/pipeline/threadDummyImpl.I	2022-01-25 14:21:26.064537897 +0100
+++ webgl-port/panda/src/pipeline/threadDummyImpl.I	2022-02-08 17:22:24.262634642 +0100
@@ -124,3 +124,13 @@
 INLINE void ThreadDummyImpl::
 consider_yield() {
 }
+
+#ifdef ANDROID
+/**
+ * Returns the JNIEnv object for the current thread.
+ */
+INLINE JNIEnv *ThreadDummyImpl::
+get_jni_env() const {
+  return nullptr;
+}
+#endif
