--- panda3d/panda/metalibs/pandagl/CMakeLists.txt	2020-03-20 15:48:21.000000000 +0100
+++ panda3d-android/panda/metalibs/pandagl/CMakeLists.txt	2020-04-13 08:32:48.998255219 +0200
@@ -1,3 +1,11 @@
+if(ANDROID)
+  set(PANDAGL_PIPE_TYPE "viewGraphicsPipe")
+  set(PANDAGL_LINK_TARGETS p3glstuff)
+  list(APPEND PANDAGL_LINK_TARGETS p3viewdisplay_gles2)
+    return()
+endif()
+
+
 if(NOT HAVE_GL)
   return()
 endif()
@@ -16,7 +24,6 @@
   list(APPEND PANDAGL_LINK_TARGETS p3cocoadisplay)
   set(PANDAGL_PIPE_TYPE "CocoaGraphicsPipe")
   set(PANDAGL_PIPE_INCLUDE "cocoaGraphicsPipe.h")
-
 else()
   message("") # Add extra line before error
   message(SEND_ERROR
