--- panda3d/panda/src/pipeline/threadDummyImpl.I	2020-03-20 15:48:21.000000000 +0100
+++ panda3d-android/panda/src/pipeline/threadDummyImpl.I	2020-04-13 03:26:24.898320799 +0200
@@ -124,3 +124,13 @@
 INLINE void ThreadDummyImpl::
 consider_yield() {
 }
+
+#ifdef ANDROID
+/**
+ * Returns the JNIEnv object for the current thread.
+ */
+INLINE JNIEnv *ThreadDummyImpl::
+get_jni_env() const {
+  return nullptr;
+}
+#endif
