--- panda3d-webgl-port/panda/src/pipeline/threadDummyImpl.h	2022-05-10 16:33:18.000000000 +0200
+++ panda3d/panda/src/pipeline/threadDummyImpl.h	2022-06-13 04:49:06.305389154 +0200
@@ -31,6 +31,12 @@
 #include <windows.h>  // For Sleep().
 #endif
 
+#ifdef ANDROID
+typedef struct _JNIEnv JNIEnv;
+bool attach_java_vm();
+static void bind_java_thread();
+#endif
+
 /**
  * A fake thread implementation for single-threaded applications.  This simply
  * fails whenever you try to start a thread.
