--- panda3d-git/panda/src/display/graphicsOutput.cxx	2020-04-13 15:10:29.000000000 +0200
+++ panda3d-aosp/panda/src/display/graphicsOutput.cxx	2020-06-07 14:46:51.677404919 +0200
@@ -410,6 +410,10 @@
  */
 bool GraphicsOutput::
 is_active() const {
+#if __ANDROID__
+    #pragma message "GraphicsOutput::is_active() false/true ?"
+    return true;
+#endif
   if (!is_valid()) {
     return false;
   }
