--- webgl-port.git/panda/src/display/graphicsOutput.cxx	2022-01-25 14:21:22.444665485 +0100
+++ webgl-port/panda/src/display/graphicsOutput.cxx	2022-02-08 17:22:24.246635193 +0100
@@ -416,6 +416,10 @@
  */
 bool GraphicsOutput::
 is_active() const {
+#if __ANDROID__
+    #pragma message "GraphicsOutput::is_active() false/true ?"
+    return true;
+#endif
   if (!is_valid()) {
     return false;
   }
