--- panda3d/panda/src/glstuff/glGraphicsStateGuardian_src.cxx	2020-03-20 15:48:21.000000000 +0100
+++ panda3d.android/panda/src/glstuff/glGraphicsStateGuardian_src.cxx	2020-04-08 02:24:26.232049738 +0200
@@ -80,6 +80,8 @@
 using std::min;
 using std::string;
 
+#error pouet
+
 TypeHandle CLP(GraphicsStateGuardian)::_type_handle;
 
 PStatCollector CLP(GraphicsStateGuardian)::_load_display_list_pcollector("Draw:Transfer data:Display lists");
