--- panda3d-00305bccae3643dda6f469f9cfe0cc46b7e5d74b/panda/src/androiddisplay/config_androiddisplay.cxx	2020-04-13 15:10:29.000000000 +0200
+++ panda3d-android/panda/src/androiddisplay/config_androiddisplay.cxx	2020-07-01 06:58:10.023126708 +0200
@@ -21,10 +21,17 @@
 
 #include "config_display.h"
 
+#include <android/log.h>
+
+#define LOG_TAG "config_androiddisplay.cxx"
+#define LOG_INFO(...) __android_log_print(ANDROID_LOG_INFO, LOG_TAG, __VA_ARGS__)
+#define LOG_ERROR(...) __android_log_print(ANDROID_LOG_ERROR, LOG_TAG, __VA_ARGS__)
+
 Configure(config_androiddisplay);
 NotifyCategoryDef(androiddisplay, "display");
 
 ConfigureFn(config_androiddisplay) {
+  LOG_INFO("34: config_androiddisplay.cxx\n");
   init_libandroiddisplay();
 }
 
@@ -37,6 +44,8 @@
 void
 init_libandroiddisplay() {
   static bool initialized = false;
+  LOG_INFO("45: config_androiddisplay.cxx\n");
+
   if (initialized) {
     return;
   }
