--- panda3d-webgl-port/panda/src/display/graphicsOutput.cxx	2022-05-10 16:33:18.000000000 +0200
+++ panda3d/panda/src/display/graphicsOutput.cxx	2022-06-13 04:42:28.015707699 +0200
@@ -416,6 +416,10 @@
  */
 bool GraphicsOutput::
 is_active() const {
+#if __ANDROID__
+    #pragma message "GraphicsOutput::is_active() false/true ?"
+    return true;
+#endif
   if (!is_valid()) {
     return false;
   }
