--- webgl-port.git/panda/src/pipeline/threadDummyImpl.h	2022-01-25 14:21:26.064537897 +0100
+++ mobile-sandbox/panda/src/pipeline/threadDummyImpl.h	2022-02-09 21:41:17.911574148 +0100
@@ -31,6 +31,10 @@
 #include <windows.h>  // For Sleep().
 #endif
 
+#ifdef ANDROID
+typedef struct _JNIEnv JNIEnv;
+#endif
+
 /**
  * A fake thread implementation for single-threaded applications.  This simply
  * fails whenever you try to start a thread.
@@ -57,6 +61,13 @@
   INLINE static void sleep(double seconds);
   INLINE static void yield();
   INLINE static void consider_yield();
+
+#ifdef ANDROID
+  INLINE JNIEnv *get_jni_env() const;
+  bool attach_java_vm();
+  static void bind_java_thread();
+#endif
+
 };
 
 #include "threadDummyImpl.I"
