--- webgl-port.git/panda/src/viewdisplay/CMakeLists.txt	2022-02-07 10:52:42.693876655 +0100
+++ mobile-sandbox/panda/src/viewdisplay/CMakeLists.txt	2022-02-10 17:25:06.637445650 +0100
@@ -0,0 +1,116 @@
+set(P3VIEWDISPLAY_HEADERS
+  config_viewdisplay.h
+  viewGraphicsPipe.h viewGraphicsPipe.I
+  viewGraphicsStateGuardian.h viewGraphicsStateGuardian.I
+  viewGraphicsWindow.h viewGraphicsWindow.I
+
+)
+
+set(P3VIEWDISPLAY_SOURCES
+  config_viewdisplay.cxx
+  viewGraphicsPipe.cxx
+  viewGraphicsStateGuardian.cxx
+  viewGraphicsWindow.cxx
+)
+
+composite_sources(p3viewdisplay P3VIEWDISPLAY_SOURCES)
+
+if(HAVE_GLES1)
+    add_component_library(p3viewdisplay_gles1 SYMBOL BUILDING_PANDAGLES
+        INIT init_libviewdisplay config_viewdisplay.h
+        ${P3VIEWDISPLAY_HEADERS} ${P3VIEWDISPLAY_SOURCES})
+    target_compile_definitions(p3viewdisplay_gles1 PUBLIC OPENGLES_1)
+    target_link_libraries(p3viewdisplay_gles2 p3glesgsg p3glstuff EGL PKG::GLES)
+
+
+    if(NOT BUILD_METALIBS)
+        install(TARGETS p3viewdisplay_gles2
+            EXPORT OpenGLES2 COMPONENT OpenGLES2
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/panda3d
+            ARCHIVE COMPONENT OpenGLES2Devel)
+    endif()
+
+    install(FILES ${P3VIEWDISPLAY_HEADERS} COMPONENT OpenGLES2Devel DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/panda3d)
+endif()
+
+if(HAVE_GLES2)
+    add_component_library(p3viewdisplay_gles2 SYMBOL BUILDING_PANDAGLES2
+        INIT init_libviewdisplay config_viewdisplay.h
+        ${P3VIEWDISPLAY_HEADERS} ${P3VIEWDISPLAY_SOURCES})
+    target_compile_definitions(p3viewdisplay_gles2 PUBLIC OPENGLES_2)
+    # panda from p3x11display ?
+    target_link_libraries(p3viewdisplay_gles2 p3gles2gsg EGL PKG::GLES2)
+
+
+    if(NOT BUILD_METALIBS)
+        install(TARGETS p3viewdisplay_gles2
+            EXPORT OpenGLES2 COMPONENT OpenGLES2
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/panda3d
+            ARCHIVE COMPONENT OpenGLES2Devel)
+    endif()
+
+    install(FILES ${P3VIEWDISPLAY_HEADERS} COMPONENT OpenGLES2Devel DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/panda3d)
+
+endif()
+
+
+set(P3VIEWDISPLAY_HEADERS
+  config_viewdisplay.h
+  viewGraphicsPipe.h viewGraphicsPipe.I
+  viewGraphicsStateGuardian.h viewGraphicsStateGuardian.I
+  viewGraphicsWindow.h viewGraphicsWindow.I
+
+)
+
+set(P3VIEWDISPLAY_SOURCES
+  config_viewdisplay.cxx
+  viewGraphicsPipe.cxx
+  viewGraphicsStateGuardian.cxx
+  viewGraphicsWindow.cxx
+)
+
+composite_sources(p3viewdisplay P3VIEWDISPLAY_SOURCES)
+
+if(HAVE_GLES1)
+    add_component_library(p3viewdisplay_gles1 SYMBOL BUILDING_PANDAGLES
+        INIT init_libviewdisplay config_viewdisplay.h
+        ${P3VIEWDISPLAY_HEADERS} ${P3VIEWDISPLAY_SOURCES})
+    target_compile_definitions(p3viewdisplay_gles1 PUBLIC OPENGLES_1)
+    target_link_libraries(p3viewdisplay_gles2 p3glesgsg p3glstuff EGL PKG::GLES)
+
+
+    if(NOT BUILD_METALIBS)
+        install(TARGETS p3viewdisplay_gles2
+            EXPORT OpenGLES2 COMPONENT OpenGLES2
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/panda3d
+            ARCHIVE COMPONENT OpenGLES2Devel)
+    endif()
+
+    install(FILES ${P3VIEWDISPLAY_HEADERS} COMPONENT OpenGLES2Devel DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/panda3d)
+endif()
+
+if(HAVE_GLES2)
+#    add_component_library(p3viewdisplay_gles2 SYMBOL BUILDING_PANDAGLES2
+#        INIT init_libviewdisplay config_viewdisplay.h
+#        ${P3VIEWDISPLAY_HEADERS} ${P3VIEWDISPLAY_SOURCES})
+    target_compile_definitions(p3viewdisplay_gles2 PUBLIC OPENGLES_2)
+    # panda from p3x11display ?
+    target_link_libraries(p3viewdisplay_gles2 p3gles2gsg EGL PKG::GLES2)
+
+
+    if(NOT BUILD_METALIBS)
+        install(TARGETS p3viewdisplay_gles2
+            EXPORT OpenGLES2 COMPONENT OpenGLES2
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/panda3d
+            ARCHIVE COMPONENT OpenGLES2Devel)
+    endif()
+
+    install(FILES ${P3VIEWDISPLAY_HEADERS} COMPONENT OpenGLES2Devel DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/panda3d)
+
+endif()
+
+
