--- panda3d-webgl-port/panda/src/pipeline/threadDummyImpl.I	2022-05-10 16:33:18.000000000 +0200
+++ panda3d/panda/src/pipeline/threadDummyImpl.I	2022-06-13 04:42:28.017707682 +0200
@@ -132,3 +132,13 @@
 get_context_switches(size_t &, size_t &) {
   return false;
 }
+
+#ifdef ANDROID
+/**
+ * Returns the JNIEnv object for the current thread.
+ */
+INLINE JNIEnv *ThreadDummyImpl::
+get_jni_env() const {
+  return nullptr;
+}
+#endif
