--- webgl-port.git/panda/src/viewdisplay/viewGraphicsPipe.I	2022-02-07 10:52:42.697876520 +0100
+++ webgl-port/panda/src/viewdisplay/viewGraphicsPipe.I	2022-02-08 17:22:24.286633815 +0100
@@ -0,0 +1,24 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file viewGraphicsPipe.I
+ * @author pmp-p
+ * @date 2020-04-07
+ */
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file viewGraphicsPipe.I
+ * @author pmp-p
+ * @date 2020-04-07
+ */
