--- panda3d/panda/src/gles2gsg/CMakeLists.txt	2020-03-20 15:48:21.000000000 +0100
+++ panda3d-android/panda/src/gles2gsg/CMakeLists.txt	2020-04-13 00:17:30.437728566 +0200
@@ -13,10 +13,13 @@
 )
 
 composite_sources(p3gles2gsg P3GLES2GSG_SOURCES)
-add_component_library(p3gles2gsg SYMBOL BUILDING_PANDAGLES2
-  ${P3GLE2SGSG_HEADERS} ${P3GLES2GSG_SOURCES})
-target_link_libraries(p3gles2gsg p3glstuff panda
-  PKG::GLES2)
+add_component_library(p3gles2gsg SYMBOL BUILDING_PANDAGLES2 ${P3GLE2SGSG_HEADERS} ${P3GLES2GSG_SOURCES})
+
+if (ANDROID)
+    target_link_libraries(p3gles2gsg p3glstuff panda PKG::GLES2)
+else()
+    target_link_libraries(p3gles2gsg p3glstuff panda PKG::GLES2)
+endif()
 
 if(NOT BUILD_METALIBS)
   install(TARGETS p3gles2gsg
