--- webgl-port.git/panda/src/physics/physicalNode.cxx	2022-02-07 20:43:57.194531078 +0100
+++ webgl-port/panda/src/physics/physicalNode.cxx	2022-02-09 18:34:05.785039390 +0100
@@ -16,7 +16,7 @@
 #include "patomic.h"
 
 // static stuff.
-static patomic_flag warned_copy_physical_node = ATOMIC_FLAG_INIT;
+static std::atomic_flag warned_copy_physical_node = ATOMIC_FLAG_INIT;
 
 TypeHandle PhysicalNode::_type_handle;
 
