--- cpython-3.8.3/Modules/readline.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-wasm/Modules/readline.c	2020-06-07 12:53:04.599798635 +0200
@@ -1175,6 +1175,11 @@
 
     completed_input_string = not_done_reading;
 
+    #ifdef __EMSCRIPTEN__
+        #error no no no i won't go there
+    #endif
+
+
     while (completed_input_string == not_done_reading) {
         int has_input = 0, err = 0;
 
