--- cpython-3.8.3/Python/pylifecycle.c	2020-05-13 19:31:54.000000000 +0200
+++ cpython-3.8.3-wasm/Python/pylifecycle.c	2020-06-07 12:53:04.615798413 +0200
@@ -2369,6 +2369,8 @@
     if (sigaction(sig, NULL, &context) == -1)
         return SIG_ERR;
     return context.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t handler;
 /* Special signal handling for the secure CRT in Visual Studio 2005 */
@@ -2415,6 +2417,8 @@
     if (sigaction(sig, &context, &ocontext) == -1)
         return SIG_ERR;
     return ocontext.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t oldhandler;
     oldhandler = signal(sig, handler);
