--- Python-3.9.12/Lib/os.py	2022-03-23 22:12:08.000000000 +0100
+++ python3-aosp/Lib/os.py	2022-04-14 08:05:00.491004814 +0200
@@ -1114,3 +1114,12 @@
             cookie,
             nt._remove_dll_directory
         )
+
+try:
+    getpid
+except:
+    import sys
+    print("210: os.getpid() is broken",file=sys.stderr)
+    def getpid():
+        return int(open('/proc/self/stat').read().split(' ')[0])
+
