--- Python-3.9.12/Modules/signalmodule.c	2022-03-23 22:12:08.000000000 +0100
+++ python3-aosp/Modules/signalmodule.c	2022-04-14 08:05:00.516004698 +0200
@@ -49,6 +49,13 @@
 #define SIG_ERR ((PyOS_sighandler_t)(-1))
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    #define SIGRTMIN 32
+    #define SIGRTMAX _NSIG
+#endif
+//PMPP API<21
+
 #ifndef NSIG
 # if defined(_NSIG)
 #  define NSIG _NSIG            /* For BSD/SysV */
