--- Python-3.9.12/Modules/_localemodule.c	2022-03-23 22:12:08.000000000 +0100
+++ python3-aosp/Modules/_localemodule.c	2022-04-14 08:05:00.505004749 +0200
@@ -108,6 +108,10 @@
                         "invalid locale category");
         return NULL;
     }
+#else
+#ifdef __ANDROID__
+    return PyUnicode_FromFormat("%s", "C");
+#endif
 #endif
 
     if (locale) {
@@ -237,7 +241,16 @@
     }
 
     /* if LC_NUMERIC is different in the C library, use saved value */
-    lc = localeconv();
+//PMPP API<21
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+    /* Don't even try on Android's broken locale.h. */
+    /* maybe can use c++ stdlib to implement */
+     lc = broken_localeconv();
+#else
+     /* if LC_NUMERIC is different in the C library, use saved value */
+     lc = localeconv(); //PATCHED
+#endif
+//PMPP API<21
 
     /* hopefully, the localeconv result survives the C library calls
        involved herein */
