--- Python-3.9.12/Python/dynload_shlib.c	2022-03-23 22:12:08.000000000 +0100
+++ python3-aosp/Python/dynload_shlib.c	2022-04-14 08:05:00.523004666 +0200
@@ -35,6 +35,9 @@
 */
 
 const char *_PyImport_DynLoadFiletab[] = {
+#if __ANDROID__
+    "." SOABI ".so",
+#else
 #ifdef __CYGWIN__
     ".dll",
 #else  /* !__CYGWIN__ */
@@ -45,6 +48,7 @@
     ".abi" PYTHON_ABI_STRING ".so",
     ".so",
 #endif  /* __CYGWIN__ */
+#endif  /* __ANDROID__ */
     NULL,
 };
 
