--- Python-3.9.12/Python/fileutils.c	2022-03-23 22:12:08.000000000 +0100
+++ python3-aosp/Python/fileutils.c	2022-04-14 08:05:00.525004656 +0200
@@ -62,6 +62,16 @@
 }
 
 
+//PMPP API<21
+#ifdef __ANDROID__
+    #if __ANDROID_API__ < 21
+        #ifndef CRYSTAX
+            #define NO_LOCALECONV 1
+        #endif
+    #endif
+#endif
+//PMPP API<21
+
 PyObject *
 _Py_device_encoding(int fd)
 {
@@ -2274,3 +2284,5 @@
 
 #undef GET_LOCALE_STRING
 }
+
+
