--- Python-3.9.1/Python/formatter_unicode.c	2020-12-07 15:02:38.000000000 +0100
+++ python3-aosp/Python/formatter_unicode.c	2020-12-08 18:29:02.626758967 +0100
@@ -692,6 +692,7 @@
 
 static const char no_grouping[1] = {CHAR_MAX};
 
+
 /* Find the decimal point character(s?), thousands_separator(s?), and
    grouping description, either for the current locale if type is
    LT_CURRENT_LOCALE, a hard-coded locale if LT_DEFAULT_LOCALE or
