--- Python-3.7.12/configure	2021-09-04 05:49:21.000000000 +0200
+++ python3-wasm/configure	2022-03-30 18:15:01.096392617 +0200
@@ -3257,6 +3257,12 @@
 	*-*-linux*)
 		ac_sys_system=Linux
 		;;
+	wasm32-*-emscripten)
+		ac_sys_system=emscripten
+		;;
+	wasm32-unknown-wasi)
+		ac_sys_system=wasi32
+		;;
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
@@ -3304,6 +3310,12 @@
 	*-*-cygwin*)
 		_host_cpu=
 		;;
+	wasm32-*-emscripten)
+		ac_sys_system=emscripten
+		;;
+	wasm32-unknown-wasi)
+		ac_sys_system=wasi32
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -9314,8 +9326,9 @@
 # -- usually .so, .sl on HP-UX, .dll on Cygwin
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking the extension of shared libraries" >&5
 $as_echo_n "checking the extension of shared libraries... " >&6; }
+
 if test -z "$SHLIB_SUFFIX"; then
-	case $ac_sys_system in
+    case $ac_sys_system in
 	hp*|HP*)
 		case `uname -m` in
 			ia64) SHLIB_SUFFIX=.so;;
@@ -9326,6 +9339,12 @@
 	*)	   SHLIB_SUFFIX=.so;;
 	esac
 fi
+
+case $ac_sys_system in
+emscripten)
+    SHLIB_SUFFIX=.wasm;;
+esac
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $SHLIB_SUFFIX" >&5
 $as_echo "$SHLIB_SUFFIX" >&6; }
 
@@ -10785,7 +10804,7 @@
 
 else
 
-    as_fn_error $? "could not find pthreads on your system" "$LINENO" 5
+    echo 'as_fn_error $? "could not find pthreads on your system" "$LINENO" 5'
 
 fi
 
@@ -11707,6 +11726,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __emscripten__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -11732,6 +11754,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __emscripten__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -12381,7 +12406,9 @@
 #ifdef HAVE_GRP_H
 #include <grp.h>
 #endif
-
+#ifdef __emscripten__
+    #error setgroups
+#endif
 int
 main ()
 {
@@ -13063,7 +13090,9 @@
 #else
 #include <sys/types.h>
 #endif
-
+#ifdef __emscripten__
+    #error makedev
+#endif
 int
 main ()
 {
@@ -15283,6 +15312,10 @@
 case $ac_sys_system in
     Linux*|GNU*|Darwin)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+    emscripten)
+        SOABI="cpython-${VERSION}-emscripten"
+        EXT_SUFFIX=".wasm"
+        ;;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac
