--- Python-3.7.10/Include/pyport.h	2021-02-16 02:29:22.000000000 +0100
+++ python3-wasm/Include/pyport.h	2021-05-04 18:24:59.667254639 +0200
@@ -696,7 +696,11 @@
  * rather than waiting for integer multiplication to trigger bogus
  * overflows.
  */
+#ifndef __EMSCRIPTEN__
 #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
+#else
+#warning "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
+#endif
 #endif
 
 #ifdef __cplusplus
