--- Python-3.7.12/Modules/_ctypes/ctypes.h	2021-09-04 05:49:21.000000000 +0200
+++ python3-wasm/Modules/_ctypes/ctypes.h	2022-03-30 18:15:01.142392342 +0200
@@ -53,7 +53,15 @@
     PyObject *b_objects;        /* dictionary of references we need to keep, or Py_None */
     union value b_value;
 };
+#if FFI_CLOSURES
+#else
 
+    #define ffi_closure void
+    #define ffi_closure_free(writable) ""
+    #define ffi_closure_alloc(size, code) ""
+    #define ffi_prep_closure_loc(...) ""
+
+#endif
 typedef struct {
     PyObject_VAR_HEAD
     ffi_closure *pcl_write; /* the C callable, writeable */
