--- Python-3.7.12/Modules/posixmodule.c	2021-09-04 05:49:21.000000000 +0200
+++ python3-wasm/Modules/posixmodule.c	2022-03-30 18:22:07.137833833 +0200
@@ -154,6 +154,12 @@
 #  endif
 #endif /* MS_WINDOWS */
 
+#if GOSH_MOVE_THEM
+    #pragma message "XXX Gosh I wish these were all moved into pyconfig.h"
+    #ifdef HAVE_FORKPTY
+    #undef HAVE_FORKPTY
+    #endif
+#else
 /* Various compilers have only certain posix functions */
 /* XXX Gosh I wish these were all moved into pyconfig.h */
 #if defined(__WATCOMC__) && !defined(__QNX__)           /* Watcom compiler */
@@ -193,7 +199,7 @@
 #define HAVE_TTYNAME    1
 #endif  /* _MSC_VER */
 #endif  /* ! __WATCOMC__ || __QNX__ */
-
+#endif // my GOSH
 
 /*[clinic input]
 # one of the few times we lie about this name!
@@ -736,7 +742,7 @@
 #define _PyLong_FromDev PyLong_FromLongLong
 
 
-#if defined(HAVE_MKNOD) && defined(HAVE_MAKEDEV)
+#if (defined(HAVE_MKNOD) && defined(HAVE_MAKEDEV)) || defined(__EMSCRIPTEN__)
 static int
 _Py_Dev_Converter(PyObject *obj, void *p)
 {
