--- Python-3.7.10/Python/dup2.c	2021-02-16 02:29:22.000000000 +0100
+++ python3-wasm/Python/dup2.c	2021-05-04 18:24:59.704254259 +0200
@@ -16,16 +16,27 @@
 
 #define BADEXIT -1
 
-int
-dup2(int fd1, int fd2)
-{
-    if (fd1 != fd2) {
-        if (fcntl(fd1, F_GETFL) < 0)
-            return BADEXIT;
-        if (fcntl(fd2, F_GETFL) >= 0)
-            close(fd2);
-        if (fcntl(fd1, F_DUPFD, fd2) < 0)
-            return BADEXIT;
+#pragma message "extern int dup2(int fd1, int fd2) ?"
+
+
+#if HAVE_DUP2
+    extern int dup2(int fd1, int fd2);
+#else
+    #if defined(__EMSCRIPTEN__)
+        #error "dup2.c will collide with the libc one"
+    #endif
+    int
+    dup2(int fd1, int fd2)
+    {
+        if (fd1 != fd2) {
+            if (fcntl(fd1, F_GETFL) < 0)
+                return BADEXIT;
+            if (fcntl(fd2, F_GETFL) >= 0)
+                close(fd2);
+            if (fcntl(fd1, F_DUPFD, fd2) < 0)
+                return BADEXIT;
+        }
+        return fd2;
     }
-    return fd2;
-}
+#endif
+
