--- Python-3.7.12/Python/fileutils.c	2021-09-04 05:49:21.000000000 +0200
+++ python3-wasm/Python/fileutils.c	2022-03-30 18:15:01.233391798 +0200
@@ -994,6 +994,9 @@
 static int
 set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
 {
+#ifdef __EMSCRIPTEN__
+    return 0;
+#else
 #ifdef MS_WINDOWS
     HANDLE handle;
     DWORD flags;
@@ -1117,6 +1120,7 @@
     }
     return 0;
 #endif
+#endif
 }
 
 /* Make the file descriptor non-inheritable.
