--- Python-3.8.1rc1/Lib/test/test_zlib.py	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Lib/test/test_zlib.py	2019-12-13 12:32:30.047977438 +0100
@@ -438,7 +438,14 @@
         sync_opt = ['Z_NO_FLUSH', 'Z_SYNC_FLUSH', 'Z_FULL_FLUSH',
                     'Z_PARTIAL_FLUSH']
 
-        ver = tuple(int(v) for v in zlib.ZLIB_RUNTIME_VERSION.split('.'))
+        v = zlib.ZLIB_RUNTIME_VERSION.split('-', 1)[0].split('.')
+        if len(v) < 4:
+            v.append('0')
+        elif not v[-1].isnumeric():
+            v[-1] = '0'
+
+        ver = tuple(map(int, v))
+
         # Z_BLOCK has a known failure prior to 1.2.5.3
         if ver >= (1, 2, 5, 3):
             sync_opt.append('Z_BLOCK')
