--- Python-3.8.1rc1/Lib/importlib/_bootstrap_external.py	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Lib/importlib/_bootstrap_external.py	2019-12-13 12:08:57.663586738 +0100
@@ -1390,6 +1390,7 @@
     refreshed when the directory the finder is handling has been modified.
 
     """
+    PATCHED = 1
 
     def __init__(self, path, *loader_details):
         """Initialize with the path to search on and a variable number of
@@ -1461,9 +1462,14 @@
                 # If a namespace package, return the path if we don't
                 #  find a module in the next section.
                 is_namespace = _path_isdir(base_path)
+
         # Check for a file w/ a proper suffix exists.
         for suffix, loader_class in self._loaders:
             full_path = _path_join(self.path, tail_module + suffix)
+            try:
+                print("1394: trying", full_path)
+            except:
+                pass
             _bootstrap._verbose_message('trying {}', full_path, verbosity=2)
             if cache_module + suffix in cache:
                 if _path_isfile(full_path):
