--- Python-3.8.1rc1/Modules/posixmodule.c	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Modules/posixmodule.c	2019-12-13 12:38:28.824555700 +0100
@@ -63,7 +63,6 @@
 disguised Unix interface).  Refer to the library manual and\n\
 corresponding Unix manual entries for more information on calls.");
 
-
 #ifdef HAVE_SYS_UIO_H
 #include <sys/uio.h>
 #endif
@@ -497,6 +496,11 @@
 #endif
 }
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    //extern ssize_t sendfile(int out_fd, int in_fd, off_t *offset, size_t count);
+#endif
+//PMPP API<21
 
 #ifdef MS_WINDOWS
 /* defined in fileutils.c */
@@ -6567,6 +6571,51 @@
 for both the master and slave ends.
 [clinic start generated code]*/
 
+#ifdef __ANDROID__
+    static PyObject *
+    os_openpty_impl(PyObject *module)
+    {
+        int master_fd = -1, slave_fd = -1;
+        char * slave_name;
+        PyOS_sighandler_t sig_saved;
+
+        master_fd = open("/dev/ptmx", O_RDWR | O_NONBLOCK | O_NOCTTY); /* open master */
+        if (master_fd < 0)
+            goto posix_error;
+
+        sig_saved = PyOS_setsig(SIGCHLD, SIG_DFL);
+
+        if (unlockpt(master_fd) < 0) {
+            PyOS_setsig(SIGCHLD, sig_saved);
+            goto posix_error;
+        }
+
+        PyOS_setsig(SIGCHLD, sig_saved);
+
+        slave_name = ptsname(master_fd); /* get name of slave */
+        if (slave_name == NULL)
+            goto posix_error;
+
+        slave_fd = _Py_open(slave_name, O_RDWR | O_NOCTTY); /* open slave */
+        if (slave_fd == -1)
+            goto error;
+
+        if (_Py_set_inheritable(master_fd, 0, NULL) < 0)
+            goto posix_error;
+
+        return Py_BuildValue("(ii)", master_fd, slave_fd);
+
+    posix_error:
+        posix_error();
+    error:
+        if (master_fd != -1)
+            close(master_fd);
+        if (slave_fd != -1)
+            close(slave_fd);
+        return NULL;
+    }
+#else
+
 static PyObject *
 os_openpty_impl(PyObject *module)
 /*[clinic end generated code: output=98841ce5ec9cef3c input=f3d99fd99e762907]*/
@@ -6655,7 +6704,7 @@
     return NULL;
 }
 #endif /* defined(HAVE_OPENPTY) || defined(HAVE__GETPTY) || defined(HAVE_DEV_PTMX) */
-
+#endif
 
 #ifdef HAVE_FORKPTY
 /*[clinic input]
@@ -9831,7 +9880,11 @@
             errno = result;
     }
 #else
+    #if __ANDROID_API__ < 21
+        result = ftruncate(path->narrow, length);
+    #else
     result = truncate(path->narrow, length);
+    #endif
 #endif
     _Py_END_SUPPRESS_IPH
     Py_END_ALLOW_THREADS
