--- Python-3.8.1rc1/Modules/_decimal/libmpdec/io.c	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Modules/_decimal/libmpdec/io.c	2019-12-13 12:32:44.059531630 +0100
@@ -871,9 +871,15 @@
         if (*spec->sep) {
             return 0;
         }
+
         spec->type = *cp++;
         spec->type = (spec->type == 'N') ? 'G' : 'g';
-        lc = localeconv();
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+        #warning "====================================== broken_localeconv ================================"
+        lc = broken_localeconv();
+#else
+        lc = localeconv(); //PATCHED
+#endif
         spec->dot = lc->decimal_point;
         spec->sep = lc->thousands_sep;
         spec->grouping = lc->grouping;
