--- Python-3.8.1rc1/Modules/signalmodule.c	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Modules/signalmodule.c	2019-12-13 12:39:43.858165351 +0100
@@ -44,6 +44,13 @@
 #define SIG_ERR ((PyOS_sighandler_t)(-1))
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    #define SIGRTMIN 32
+    #define SIGRTMAX _NSIG
+#endif
+//PMPP API<21
+
 #ifndef NSIG
 # if defined(_NSIG)
 #  define NSIG _NSIG            /* For BSD/SysV */
