--- Python-3.8.1rc1/Modules/mmapmodule.c	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Modules/mmapmodule.c	2019-12-13 12:38:23.988709729 +0100
@@ -79,6 +79,12 @@
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
+
 typedef enum
 {
     ACCESS_DEFAULT,
