--- Python-3.8.1rc1/Python/importdl.c	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Python/importdl.c	2019-12-13 12:44:33.582930555 +0100
@@ -131,6 +131,9 @@
     pathbytes = PyUnicode_EncodeFSDefault(path);
     if (pathbytes == NULL)
         goto error;
+#if __ANDROID__
+#warning "FIXME: supress lib from "PyInit_lib"
+#endif
     exportfunc = _PyImport_FindSharedFuncptr(hook_prefix, name_buf,
                                              PyBytes_AS_STRING(pathbytes),
                                              fp);
