--- Python-3.8.1rc1/Python/formatter_unicode.c	2019-12-09 18:47:55.000000000 +0100
+++ Python-3.8.1rc1.pydk/Python/formatter_unicode.c	2019-12-13 12:44:19.741371918 +0100
@@ -690,6 +690,7 @@
 
 static const char no_grouping[1] = {CHAR_MAX};
 
+
 /* Find the decimal point character(s?), thousands_separator(s?), and
    grouping description, either for the current locale if type is
    LT_CURRENT_LOCALE, a hard-coded locale if LT_DEFAULT_LOCALE or
