--- cpython-3.8.5/Modules/signalmodule.c	2020-07-20 15:01:32.000000000 +0200
+++ cpython-3.8.5-aosp/Modules/signalmodule.c	2020-08-27 18:54:48.136970307 +0200
@@ -44,6 +44,13 @@
 #define SIG_ERR ((PyOS_sighandler_t)(-1))
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    #define SIGRTMIN 32
+    #define SIGRTMAX _NSIG
+#endif
+//PMPP API<21
+
 #ifndef NSIG
 # if defined(_NSIG)
 #  define NSIG _NSIG            /* For BSD/SysV */
