--- cpython-3.8.5/Modules/mmapmodule.c	2020-07-20 15:01:32.000000000 +0200
+++ cpython-3.8.5-aosp/Modules/mmapmodule.c	2020-08-27 18:54:48.129970390 +0200
@@ -79,6 +79,12 @@
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
+
 typedef enum
 {
     ACCESS_DEFAULT,
