--- python3/Modules/_decimal/libmpdec/io.c	2019-10-15 00:32:36.000000000 +0200
+++ python3-android/Modules/_decimal/libmpdec/io.c	2019-10-23 18:14:07.999612800 +0200
@@ -871,9 +871,15 @@
         if (*spec->sep) {
             return 0;
         }
+
         spec->type = *cp++;
         spec->type = (spec->type == 'N') ? 'G' : 'g';
-        lc = localeconv();
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+        #warning "====================================== broken_localeconv ================================"
+        lc = broken_localeconv();
+#else
+        lc = localeconv(); //PATCHED
+#endif
         spec->dot = lc->decimal_point;
         spec->sep = lc->thousands_sep;
         spec->grouping = lc->grouping;
