--- python3/Python/fileutils.c	2019-10-15 00:32:36.000000000 +0200
+++ python3-android/Python/fileutils.c	2019-10-23 18:14:08.015612282 +0200
@@ -32,6 +32,16 @@
 int _Py_open_cloexec_works = -1;
 #endif
 
+//PMPP API<21
+#ifdef __ANDROID__
+    #if __ANDROID_API__ < 21
+        #ifndef CRYSTAX
+            #define NO_LOCALECONV 1
+        #endif
+    #endif
+#endif
+//PMPP API<21
+
 PyObject *
 _Py_device_encoding(int fd)
 {
@@ -1863,3 +1873,5 @@
     PyMem_Free(oldloc);
     return res;
 }
+
+
