--- python3/Python/clinic/import.c.h	2019-10-15 00:32:36.000000000 +0200
+++ python3-android/Python/clinic/import.c.h	2019-10-24 05:58:15.487026488 +0200
@@ -286,6 +286,9 @@
         &spec, &file)) {
         goto exit;
     }
+#if __ANDROID__
+    printf("290: _imp_create_dynamic: %s\n",PyUnicode_AsUTF8(PyObject_GetAttrString(spec, "name")));
+#endif
     return_value = _imp_create_dynamic_impl(module, spec, file);
 
 exit:
