--- python3/Python/import.c	2019-10-15 00:32:36.000000000 +0200
+++ python3-android/Python/import.c	2019-10-24 05:18:15.618168665 +0200
@@ -2147,7 +2147,9 @@
         Py_DECREF(name);
         return NULL;
     }
-
+#if __ANDROID__
+    printf("2151: name: %s, path:%s\n",PyUnicode_AsUTF8(name),PyUnicode_AsUTF8(path));
+#endif
     mod = _PyImport_FindExtensionObject(name, path);
     if (mod != NULL || PyErr_Occurred()) {
         Py_DECREF(name);
