--- Python-3.9.1/configure	2020-12-07 15:02:38.000000000 +0100
+++ Python-3.9.1-wasm/configure	2020-12-08 08:51:13.990216362 +0100
@@ -3301,6 +3301,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	wasm-*-*)
+		_host_cpu=
+		;;
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
@@ -3354,6 +3357,9 @@
 	*-*-vxworks*)
 		_host_cpu=$host_cpu
 		;;
+	wasm-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -9438,8 +9444,9 @@
 # -- usually .so, .sl on HP-UX, .dll on Cygwin
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking the extension of shared libraries" >&5
 $as_echo_n "checking the extension of shared libraries... " >&6; }
+
 if test -z "$SHLIB_SUFFIX"; then
-	case $ac_sys_system in
+    case $ac_sys_system in
 	hp*|HP*)
 		case `uname -m` in
 			ia64) SHLIB_SUFFIX=.so;;
@@ -9450,6 +9457,12 @@
 	*)	   SHLIB_SUFFIX=.so;;
 	esac
 fi
+
+case $ac_sys_system in
+Emscripten)
+    SHLIB_SUFFIX=.wasm;;
+esac
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $SHLIB_SUFFIX" >&5
 $as_echo "$SHLIB_SUFFIX" >&6; }
 
@@ -10968,7 +10981,7 @@
 
 else
 
-    as_fn_error $? "could not find pthreads on your system" "$LINENO" 5
+    echo 'as_fn_error $? "could not find pthreads on your system" "$LINENO" 5'
 
 fi
 
@@ -11921,6 +11934,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -11946,6 +11962,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -12653,7 +12672,9 @@
 #ifdef HAVE_GRP_H
 #include <grp.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error setgroups
+#endif
 int
 main ()
 {
@@ -13304,7 +13325,9 @@
 #else
 #include <sys/types.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error makedev
+#endif
 int
 main ()
 {
@@ -15386,6 +15409,10 @@
 case $ac_sys_system in
     Linux*|GNU*|Darwin|VxWorks)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+    Emscripten)
+        SOABI="cpython-${VERSION}-emscripten"
+        EXT_SUFFIX=".wasm"
+        ;;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac
