--- cpython-3.8.5/Python/dynload_shlib.c	2020-07-20 15:01:32.000000000 +0200
+++ cpython-3.8.5-aosp/Python/dynload_shlib.c	2020-08-27 18:54:48.141970248 +0200
@@ -34,6 +34,9 @@
 */
 
 const char *_PyImport_DynLoadFiletab[] = {
+#if __ANDROID__
+    "." SOABI ".so",
+#else
 #ifdef __CYGWIN__
     ".dll",
 #else  /* !__CYGWIN__ */
@@ -44,6 +47,7 @@
     ".abi" PYTHON_ABI_STRING ".so",
     ".so",
 #endif  /* __CYGWIN__ */
+#endif  /* __ANDROID__ */
     NULL,
 };
 
