--- cpython-3.8.5/Tools/scripts/update_file.py	2020-07-20 15:01:32.000000000 +0200
+++ cpython-3.8.5-aosp/Tools/scripts/update_file.py	2020-08-27 18:54:48.154970094 +0200
@@ -11,8 +11,11 @@
 
 
 def main(old_path, new_path):
-    with open(old_path, 'rb') as f:
-        old_contents = f.read()
+    if os.path.exists(old_path):
+        with open(old_path, 'rb') as f:
+            old_contents = f.read()
+    else:
+        old_contents =""
     with open(new_path, 'rb') as f:
         new_contents = f.read()
     if old_contents != new_contents:
