--- panda3d-1.10.0/panda/src/device/p3device_composite1.cxx	2019-01-03 16:01:10.000000000 +0100
+++ panda3d-1.10.0.armv7a-androideabi/panda/src/device/p3device_composite1.cxx	2019-01-14 07:48:06.285357752 +0100
@@ -1,3 +1,17 @@
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 20
+    #include <linux/input.h>
+    #define EV_CNT (EV_MAX+1)
+    #define KEY_CNT (KEY_MAX+1)
+    #define ABS_CNT (ABS_MAX+1)
+    #define FF_CNT (FF_MAX+1)
+    #define BTN_TRIGGER_HAPPY1 0x2c0
+    #define BTN_TRIGGER_HAPPY2 0x2c1
+    #define BTN_TRIGGER_HAPPY3 0x2c2
+    #define BTN_TRIGGER_HAPPY4 0x2c3
+
+    #define IN_CLOEXEC O_CLOEXEC
+    #define IN_NONBLOCK O_NONBLOCK
+#endif
 
 #include "clientAnalogDevice.cxx"
 #include "clientBase.cxx"
