--- Python-3.10.4/Lib/test/test_zlib.py	2022-03-23 21:12:04.000000000 +0100
+++ python3-aosp/Lib/test/test_zlib.py	2022-04-15 04:43:20.121954215 +0200
@@ -445,7 +445,14 @@
         sync_opt = ['Z_NO_FLUSH', 'Z_SYNC_FLUSH', 'Z_FULL_FLUSH',
                     'Z_PARTIAL_FLUSH']
 
-        ver = tuple(int(v) for v in zlib.ZLIB_RUNTIME_VERSION.split('.'))
+        v = zlib.ZLIB_RUNTIME_VERSION.split('-', 1)[0].split('.')
+        if len(v) < 4:
+            v.append('0')
+        elif not v[-1].isnumeric():
+            v[-1] = '0'
+
+        ver = tuple(map(int, v))
+
         # Z_BLOCK has a known failure prior to 1.2.5.3
         if ver >= (1, 2, 5, 3):
             sync_opt.append('Z_BLOCK')
