--- Python-3.10.5/Parser/pegen.c	2022-06-06 13:53:30.000000000 +0200
+++ Python-3.10.5-pydk/Parser/pegen.c	2022-06-13 04:10:09.549166880 +0200
@@ -432,7 +432,7 @@
      * (multi-line) statement are stored in p->tok->interactive_src_start.
      * If not, we're parsing from a string, which means that the whole source
      * is stored in p->tok->str. */
-    assert((p->tok->fp == NULL && p->tok->str != NULL) || p->tok->fp == stdin);
+    assert((p->tok->fp == NULL && p->tok->str != NULL) || p->tok->fp != NULL);
 
     char *cur_line = p->tok->fp_interactive ? p->tok->interactive_src_start : p->tok->str;
     if (cur_line == NULL) {
