--- cpython-bpo-30386/Android/build-config	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.19/Android/build-config	2017-12-12 22:45:46.677471595 +0100
@@ -21,7 +21,7 @@
     build_config_error "The installed NDK version is $version but version $ndk_version is required."
 unset ndk_version version
 
-test $((ANDROID_API)) -lt 21 && \
+test $((ANDROID_API)) -lt 19 && \
     build_config_error "\$ANDROID_API ($ANDROID_API) must be 21 or greater."
 test $((ANDROID_API)) -lt 24 && test "$ANDROID_ARCH" = arm64 && \
     build_config_error "Cannot build or run arm64 at an API level less than 24."
