--- cpython-bpo-30386/Modules/termios.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.api19/Modules/termios.c	2018-02-01 05:04:01.091120257 +0100
@@ -379,7 +379,11 @@
 #endif
 
     /* tcsetattr() constants */
+#if defined(__ANDROID_API__) && __ANDROID_API__ > 0
+    {"TCSANOW", TCSETS},     // https://github.com/android-ndk/ndk/issues/441
+#else
     {"TCSANOW", TCSANOW},
+#endif
     {"TCSADRAIN", TCSADRAIN},
     {"TCSAFLUSH", TCSAFLUSH},
 #ifdef TCSASOFT
