--- cpython-bpo-30386/Modules/posixmodule.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.api19/Modules/posixmodule.c	2018-01-26 11:31:32.000000000 +0100
@@ -9,7 +9,6 @@
    test macro, e.g. '_MSC_VER'. */
 
 
-
 #ifdef __APPLE__
    /*
     * Step 1 of support for weak-linking a number of symbols existing on
@@ -394,6 +393,11 @@
 #define MODNAME "posix"
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern ssize_t sendfile(int out_fd, int in_fd, off_t *offset, size_t count);
+#endif
+//PMPP API<21
 
 #ifdef HAVE_FORK
 static void
@@ -2596,7 +2600,18 @@
 #define X_OK 1
 #endif
 
-
+#if __ANDROID__
+    #define HAVE_TTYNAME 1
+    static char *
+    os_ttyname_impl(PyObject *module, int fd)
+    {
+        char *ret;
+        ret = ttyname(fd);
+        if (ret == NULL)
+            posix_error();
+        return ret;
+    }
+#else
 #ifdef HAVE_TTYNAME
 /*[clinic input]
 os.ttyname -> DecodeFSDefault
@@ -2621,6 +2636,7 @@
     return ret;
 }
 #endif
+#endif
 
 #ifdef HAVE_CTERMID
 /*[clinic input]
@@ -5883,6 +5899,51 @@
 for both the master and slave ends.
 [clinic start generated code]*/
 
+#ifdef __ANDROID__
+    static PyObject *
+    os_openpty_impl(PyObject *module)
+    {
+        int master_fd = -1, slave_fd = -1;
+        char * slave_name;
+        PyOS_sighandler_t sig_saved;
+
+        master_fd = open("/dev/ptmx", O_RDWR | O_NONBLOCK | O_NOCTTY); /* open master */
+        if (master_fd < 0)
+            goto posix_error;
+
+        sig_saved = PyOS_setsig(SIGCHLD, SIG_DFL);
+
+        if (unlockpt(master_fd) < 0) {
+            PyOS_setsig(SIGCHLD, sig_saved);
+            goto posix_error;
+        }
+
+        PyOS_setsig(SIGCHLD, sig_saved);
+
+        slave_name = ptsname(master_fd); /* get name of slave */
+        if (slave_name == NULL)
+            goto posix_error;
+
+        slave_fd = _Py_open(slave_name, O_RDWR | O_NOCTTY); /* open slave */
+        if (slave_fd == -1)
+            goto error;
+
+        if (_Py_set_inheritable(master_fd, 0, NULL) < 0)
+            goto posix_error;
+
+        return Py_BuildValue("(ii)", master_fd, slave_fd);
+
+    posix_error:
+        posix_error();
+    error:
+        if (master_fd != -1)
+            close(master_fd);
+        if (slave_fd != -1)
+            close(slave_fd);
+        return NULL;
+    }
+#else
+
 static PyObject *
 os_openpty_impl(PyObject *module)
 /*[clinic end generated code: output=98841ce5ec9cef3c input=f3d99fd99e762907]*/
@@ -5971,7 +6032,7 @@
     return NULL;
 }
 #endif /* defined(HAVE_OPENPTY) || defined(HAVE__GETPTY) || defined(HAVE_DEV_PTMX) */
-
+#endif
 
 #ifdef HAVE_FORKPTY
 /*[clinic input]
