--- cpython-bpo-30386/Modules/_localemodule.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.api19/Modules/_localemodule.c	2018-01-26 11:36:10.000000000 +0100
@@ -103,6 +103,12 @@
         PyErr_SetString(Error, "invalid locale category");
         return NULL;
     }
+#else
+    #ifdef __ANDROID__
+        #if defined(__ANDROID_API__) && __ANDROID_API__ < 20
+            return PyUnicode_FromFormat("%s", "C");
+        #endif
+    #endif
 #endif
 
     if (locale) {
@@ -143,7 +149,15 @@
         return NULL;
 
     /* if LC_NUMERIC is different in the C library, use saved value */
-    l = localeconv();
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    /* Don't even try on Android's broken locale.h. */
+    goto failed;
+#else
+     /* if LC_NUMERIC is different in the C library, use saved value */
+     l = localeconv(); //PATCHED
+#endif
+//PMPP API<21
 
     /* hopefully, the localeconv result survives the C library calls
        involved herein */
