--- cpython-bpo-30386/Objects/obmalloc.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.api19/Objects/obmalloc.c	2018-01-26 11:31:32.000000000 +0100
@@ -2,6 +2,11 @@
 
 #include <stdbool.h>
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
 
 /* Defined in tracemalloc.c */
 extern void _PyMem_DumpTraceback(int fd, const void *ptr);
