--- cpython-bpo-30386/Python/formatter_unicode.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.api19/Python/formatter_unicode.c	2018-01-21 19:11:30.000000000 +0100
@@ -704,7 +704,9 @@
 {
     switch (type) {
     case LT_CURRENT_LOCALE: {
-        struct lconv *locale_data = localeconv();
+//PMPP API<21
+#if __ANDROID_API__ > 19
+        struct lconv *locale_data = localeconv(); //PATCHED
         locale_info->decimal_point = PyUnicode_DecodeLocale(
                                          locale_data->decimal_point,
                                          NULL);
@@ -717,6 +719,8 @@
             return -1;
         locale_info->grouping = locale_data->grouping;
         break;
+#endif
+//PMPP API<21
     }
     case LT_DEFAULT_LOCALE:
     case LT_UNDERSCORE_LOCALE:
