--- cpython-bpo-30386/Python/pystrtod.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.api19/Python/pystrtod.c	2018-01-26 11:31:32.000000000 +0100
@@ -176,9 +176,14 @@
     assert(nptr != NULL);
 
     fail_pos = NULL;
-
-    locale_data = localeconv();
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    decimal_point = ".";
+#else
+    locale_data = localeconv(); //PATCHED
     decimal_point = locale_data->decimal_point;
+#endif
+//PMPP API<21
     decimal_point_len = strlen(decimal_point);
 
     assert(decimal_point_len != 0);
