--- python3/Lib/hashlib.py	2020-02-24 22:36:25.000000000 +0100
+++ python3-wasm/Lib/hashlib.py	2020-04-20 18:37:21.093123833 +0200
@@ -55,10 +55,14 @@
 
 # This tuple and __get_builtin_constructor() must be modified if a new
 # always available algorithm is added.
-__always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512',
-                      'blake2b', 'blake2s',
-                      'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512',
-                      'shake_128', 'shake_256')
+import sys
+if sys.platform=='emscripten':
+    __always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')
+else:
+    __always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512',
+                          'blake2b', 'blake2s',
+                          'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512',
+                          'shake_128', 'shake_256')
 
 
 algorithms_guaranteed = set(__always_supported)
