--- python3/Modules/posixmodule.c	2020-02-24 22:36:25.000000000 +0100
+++ python3-wasm/Modules/posixmodule.c	2020-04-20 18:37:21.103123690 +0200
@@ -174,6 +174,12 @@
 #  endif
 #endif /* MS_WINDOWS */
 
+#if GOSH_MOVE_THEM
+    #pragma message "XXX Gosh I wish these were all moved into pyconfig.h"
+    #ifdef HAVE_FORKPTY
+    #undef HAVE_FORKPTY
+    #endif
+#else
 /* Various compilers have only certain posix functions */
 /* XXX Gosh I wish these were all moved into pyconfig.h */
 #if defined(__WATCOMC__) && !defined(__QNX__)           /* Watcom compiler */
@@ -215,7 +221,7 @@
 #define HAVE_TTYNAME    1
 #endif  /* _MSC_VER */
 #endif  /* ! __WATCOMC__ || __QNX__ */
-
+#endif // my GOSH
 
 /*[clinic input]
 # one of the few times we lie about this name!
@@ -5389,9 +5395,13 @@
 
    if (setsigmask) {
         sigset_t set;
+		#if __EMSCRIPTEN__
+			#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+		#else
         if (!_Py_Sigset_Converter(setsigmask, &set)) {
             goto fail;
         }
+		#endif
         errno = posix_spawnattr_setsigmask(attrp, &set);
         if (errno) {
             posix_error();
@@ -5402,9 +5412,13 @@
 
     if (setsigdef) {
         sigset_t set;
+		#if __EMSCRIPTEN__
+			#pragma message "HAVE_SIGSET_T // _Py_Sigset_Converter ?"
+		#else
         if (!_Py_Sigset_Converter(setsigdef, &set)) {
             goto fail;
         }
+		#endif
         errno = posix_spawnattr_setsigdefault(attrp, &set);
         if (errno) {
             posix_error();
