--- python3/Modules/_posixsubprocess.c	2020-02-24 22:36:25.000000000 +0100
+++ python3-wasm/Modules/_posixsubprocess.c	2020-04-20 18:37:21.107123633 +0200
@@ -552,7 +552,7 @@
     }
 }
 
-
+#ifdef HAVE_FORK
 static PyObject *
 subprocess_fork_exec(PyObject* self, PyObject *args)
 {
@@ -762,7 +762,7 @@
     Py_XDECREF(gc_module);
     return NULL;
 }
-
+#endif
 
 PyDoc_STRVAR(subprocess_fork_exec_doc,
 "fork_exec(args, executable_list, close_fds, cwd, env,\n\
@@ -793,7 +793,9 @@
 
 
 static PyMethodDef module_methods[] = {
+#ifdef HAVE_FORK
     {"fork_exec", subprocess_fork_exec, METH_VARARGS, subprocess_fork_exec_doc},
+#endif
     {NULL, NULL}  /* sentinel */
 };
 
