--- python3/Modules/selectmodule.c	2020-02-24 22:36:25.000000000 +0100
+++ python3-wasm/Modules/selectmodule.c	2020-04-20 18:37:21.111123575 +0200
@@ -322,7 +322,11 @@
     do {
         Py_BEGIN_ALLOW_THREADS
         errno = 0;
+#if defined(__EMSCRIPTEN__)
+        n = select(max, &ifdset, &ofdset, NULL, tvp); // &efdset exceptfds not supported (assert)
+#else
         n = select(max, &ifdset, &ofdset, &efdset, tvp);
+#endif
         Py_END_ALLOW_THREADS
 
         if (errno != EINTR)
