--- python3/Python/import.c	2020-02-24 22:36:25.000000000 +0100
+++ python3-wasm/Python/import.c	2020-04-17 05:50:52.724788410 +0200
@@ -2028,6 +2028,20 @@
     list = PyList_New(0);
     if (list == NULL)
         return NULL;
+
+#ifdef __EMSCRIPTEN__
+        PyObject *item = PyUnicode_FromString( ".cpython-38-em.wasm");
+        if (item == NULL) {
+            Py_DECREF(list);
+            return NULL;
+        }
+        if (PyList_Append(list, item) < 0) {
+            Py_DECREF(list);
+            Py_DECREF(item);
+            return NULL;
+        }
+        Py_DECREF(item);
+#else
 #ifdef HAVE_DYNAMIC_LOADING
     const char *suffix;
     unsigned int index = 0;
@@ -2047,6 +2061,7 @@
         index += 1;
     }
 #endif
+#endif
     return list;
 }
 
