--- cpython-bpo-30386/configure	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.em/configure	2018-03-25 17:21:27.172250548 +0200
@@ -3245,6 +3245,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	asmjs-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3289,6 +3292,9 @@
 	*-*-cygwin*)
 		_host_cpu=
 		;;
+	asmjs-*-*)
+		_host_cpu=
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
