--- cpython-bpo-30386/configure	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.em/configure	2018-04-22 18:51:44.794088934 +0200
@@ -3245,6 +3245,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	asmjs-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3289,6 +3292,9 @@
 	*-*-cygwin*)
 		_host_cpu=
 		;;
+	asmjs-*-*)
+		_host_cpu=
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -11342,6 +11348,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -11367,6 +11376,9 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <sys/epoll.h>
+#ifdef __EMSCRIPTEN__
+#error unsupported
+#endif
 int
 main ()
 {
@@ -12016,7 +12028,9 @@
 #ifdef HAVE_GRP_H
 #include <grp.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error setgroups
+#endif
 int
 main ()
 {
@@ -12554,7 +12568,9 @@
 #else
 #include <sys/types.h>
 #endif
-
+#ifdef __EMSCRIPTEN__
+    #error makedev
+#endif
 int
 main ()
 {
