--- cpython-bpo-30386/Lib/hashlib.py	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.em/Lib/hashlib.py	2018-04-22 16:52:07.051255193 +0200
@@ -55,10 +55,14 @@
 
 # This tuple and __get_builtin_constructor() must be modified if a new
 # always available algorithm is added.
-__always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512',
-                      'blake2b', 'blake2s',
-                      'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512',
-                      'shake_128', 'shake_256')
+import sys
+if sys.platform=='emscripten':
+    __always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')
+else:
+    __always_supported = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512',
+                          'blake2b', 'blake2s',
+                          'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512',
+                          'shake_128', 'shake_256')
 
 
 algorithms_guaranteed = set(__always_supported)
