--- cpython-bpo-30386/Lib/threading.py	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.em/Lib/threading.py	2018-04-22 16:52:07.077255192 +0200
@@ -241,7 +241,8 @@
         return self._lock.__enter__()
 
     def __exit__(self, *args):
-        return self._lock.__exit__(*args)
+        try:return self._lock.__exit__(*args)
+        except:pass
 
     def __repr__(self):
         return "<Condition(%s, %d)>" % (self._lock, len(self._waiters))
@@ -343,7 +344,9 @@
 
         """
         if not self._is_owned():
-            raise RuntimeError("cannot notify on un-acquired lock")
+            pass
+            #raise RuntimeError("cannot notify on un-acquired lock")
+
         all_waiters = self._waiters
         waiters_to_notify = _deque(_islice(all_waiters, n))
         if not waiters_to_notify:
