--- cpython-bpo-30386/Lib/subprocess.py	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.em/Lib/subprocess.py	2018-04-09 05:22:12.897230781 +0200
@@ -40,10 +40,9 @@
 getstatusoutput(...): Runs a command in the shell, waits for it to complete,
     then returns a (exitcode, output) tuple
 """
-
 import sys
 _mswindows = (sys.platform == "win32")
-
+_emscripten = (sys.platform == "emscripten")
 import io
 import os
 import time
@@ -135,7 +134,8 @@
             self.hStdError = hStdError
             self.wShowWindow = wShowWindow
 else:
-    import _posixsubprocess
+    if not _emscripten:
+        import _posixsubprocess
     import select
     import selectors
     import threading
