--- cpython-bpo-30386/Python/fileutils.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.em/Python/fileutils.c	2018-03-25 17:21:27.212250547 +0200
@@ -789,6 +789,9 @@
 static int
 set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
 {
+#ifdef __EMSCRIPTEN__
+    return 0;
+#else
 #ifdef MS_WINDOWS
     HANDLE handle;
     DWORD flags;
@@ -903,6 +906,7 @@
     }
     return 0;
 #endif
+#endif
 }

 /* Make the file descriptor non-inheritable.
