--- cpython-bpo-30386/Python/pylifecycle.c	2017-12-10 15:07:21.000000000 +0100
+++ cpython-bpo-30386.em/Python/pylifecycle.c	2018-04-22 16:52:07.162255190 +0200
@@ -2222,6 +2222,8 @@
     if (sigaction(sig, NULL, &context) == -1)
         return SIG_ERR;
     return context.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t handler;
 /* Special signal handling for the secure CRT in Visual Studio 2005 */
@@ -2268,6 +2270,8 @@
     if (sigaction(sig, &context, &ocontext) == -1)
         return SIG_ERR;
     return ocontext.sa_handler;
+#elif defined(__EMSCRIPTEN__)
+    return SIG_ERR;
 #else
     PyOS_sighandler_t oldhandler;
     oldhandler = signal(sig, handler);
