--- panda3d-webgl-port/panda/src/express/trueClock.cxx	2019-01-06 14:15:01.000000000 +0100
+++ panda3d-webgl-port.em/panda/src/express/trueClock.cxx	2019-01-25 06:21:18.830058841 +0100
@@ -483,47 +483,7 @@
   _time_scale = new_time_scale;
 }
 
-#elif defined(__EMSCRIPTEN__)
-
-/**
- * The Emscripten implementation.  This uses either the JavaScript function
- * performance.now() if available, otherwise Date.now().
- */
-
-#include <emscripten.h>
-
-/**
- *
- */
-double TrueClock::
-get_long_time() {
-  return emscripten_get_now() * 0.001;
-}
-
-/**
- *
- */
-double TrueClock::
-get_short_raw_time() {
-  return emscripten_get_now() * 0.001;
-}
-
-/**
- *
- */
-bool TrueClock::
-set_cpu_affinity(uint32_t mask) const {
-  return false;
-}
-
-/**
- *
- */
-TrueClock::
-TrueClock() {
-}
-
-#else  // !WIN32_VC
+#else
 
 // The Posix implementation.
 
