--- Python-3.7.12/Modules/readline.c	2021-09-04 05:49:21.000000000 +0200
+++ python3-wasm/Modules/readline.c	2022-03-30 18:15:01.175392144 +0200
@@ -1184,6 +1184,11 @@
 
     completed_input_string = not_done_reading;
 
+    #ifdef __EMSCRIPTEN__
+        #error no no no i won't go there
+    #endif
+
+
     while (completed_input_string == not_done_reading) {
         int has_input = 0, err = 0;
 
