--- Python-3.10.4/Lib/locale.py	2022-03-23 21:12:04.000000000 +0100
+++ python3-aosp/Lib/locale.py	2022-04-15 04:43:20.119954223 +0200
@@ -46,6 +46,10 @@
 
     from _locale import *
 
+    if hasattr(sys, 'getandroidapilevel'):
+        if sys.getandroidapilevel()<21:
+            raise ImportError("android-api19")
+
 except ImportError:
 
     # Locale emulation
@@ -88,6 +92,9 @@
         """ setlocale(integer,string=None) -> string.
             Activates/queries locale processing.
         """
+        if category>LC_ALL:
+            raise Error('_locale emulation invalid category')
+
         if value not in (None, '', 'C'):
             raise Error('_locale emulation only supports "C" locale')
         return 'C'
