--- Python-3.10.4/Modules/readline.c	2022-03-23 21:12:04.000000000 +0100
+++ python3-aosp/Modules/readline.c	2022-04-15 04:43:20.125954198 +0200
@@ -1324,7 +1324,9 @@
         int has_input = 0, err = 0;
 
         while (!has_input)
-        {               struct timeval timeout = {0, 100000}; /* 0.1 seconds */
+        {
+
+            struct timeval timeout = {0, 10000}; /* 0.01 seconds */
 
             /* [Bug #1552726] Only limit the pause if an input hook has been
                defined.  */
