--- Python-3.10.4/Python/formatter_unicode.c	2022-03-23 21:12:04.000000000 +0100
+++ python3-aosp/Python/formatter_unicode.c	2022-04-15 04:43:20.127954190 +0200
@@ -692,6 +692,7 @@
 
 static const char no_grouping[1] = {CHAR_MAX};
 
+
 /* Find the decimal point character(s?), thousands_separator(s?), and
    grouping description, either for the current locale if type is
    LT_CURRENT_LOCALE, a hard-coded locale if LT_DEFAULT_LOCALE or
