--- panda3d-webgl-port/direct/src/showbase/PythonUtil.py	2019-01-06 14:15:01.000000000 +0100
+++ panda3d-webgl-port.em/direct/src/showbase/PythonUtil.py	2019-02-06 17:20:41.372247140 +0100
@@ -2513,7 +2513,7 @@
         index = -1
         while True:
             curChar = self._curCounter[index]
-            if curChar is 'Z':
+            if curChar == 'Z':
                 nextChar = 'A'
                 carry = True
             else:
