--- panda3d-webgl-port/dtool/src/interrogate/interrogate_module.cxx	2018-05-07 17:07:51.000000000 +0200
+++ panda3d-webgl-port.pp/dtool/src/interrogate/interrogate_module.cxx	2018-10-27 21:37:58.315635294 +0200
@@ -315,7 +315,7 @@
       << "#if PY_MAJOR_VERSION >= 3\n"
       << "static struct PyModuleDef py_" << library_name << "_module = {\n"
       << "  PyModuleDef_HEAD_INIT,\n"
-      << "  \"" << library_name << "\",\n"
+      << "  \"" << module_name << "\",\n"
       << "  NULL,\n"
       << "  -1,\n"
       << "  NULL,\n"
