--- panda3d-webgl-port/dtool/src/interrogatedb/py_wrappers.cxx	2019-01-06 14:15:01.000000000 +0100
+++ panda3d-webgl-port.em/dtool/src/interrogatedb/py_wrappers.cxx	2019-02-04 23:24:17.874381787 +0100
@@ -3,6 +3,8 @@
  * @author rdb
  * @date 2017-11-26
  */
+#include "Python.h"
+#include "structmember.h" /* Why is this not included in Python.h? */
 
 #include "py_wrappers.h"
 
@@ -1085,13 +1087,26 @@
   return wrap->_iternext_func(wrap->_base._self);
 }
 
+
+
+
+
+
+
+
+
+
+
 /**
  * This is a variant of the Python getset mechanism that permits static
  * properties.
  */
 static void
 Dtool_StaticProperty_dealloc(PyDescrObject *descr) {
-  _PyObject_GC_UNTRACK(descr);
+//PMPP
+	#pragma message "3.8 removed  _PyObject_GC_UNTRACK(descr); "
+	PyObject_GC_UnTrack(descr);
+//PMPP
   Py_XDECREF(descr->d_type);
   Py_XDECREF(descr->d_name);
 //#if PY_MAJOR_VERSION >= 3
