--- panda3d-webgl-port/makepanda/makepandacore.py	2018-05-07 17:07:51.000000000 +0200
+++ panda3d-webgl-port.pp/makepanda/makepandacore.py	2018-10-27 21:37:58.355635293 +0200
@@ -2172,7 +2172,7 @@
     if CrossCompiling():
         # We need a version of Python we can run.
         SDK["PYTHONEXEC"] = sys.executable
-        host_version = "python" + sysconfig.get_python_version() + abiflags
+        host_version = "python" + sysconfig.get_python_version()
         if SDK["PYTHONVERSION"] != host_version:
             exit("Host Python version (%s) must be the same as target Python version (%s)!" % (host_version, SDK["PYTHONVERSION"]))
 
@@ -2887,6 +2887,8 @@
 
         null.close()
 
+        null.close()
+
         # Print out the search paths
         if GetVerbose():
             print("System library search path:")
@@ -3219,15 +3221,16 @@
     ORIG_EXT[x] = v
 
 def GetExtensionSuffix():
+    target = GetTarget()
+    if target == 'emscripten':
+        return '.bc'
     if sys.version_info >= (3, 0):
         suffix = sysconfig.get_config_var('EXT_SUFFIX')
         if suffix:
             return suffix
-    target = GetTarget()
+
     if target == 'windows':
         return '.pyd'
-    elif target == 'emscripten':
-        return '.bc'
     else:
         return '.so'
 
