--- panda3d-webgl-port/makepanda/makepandacore.py	2019-01-06 14:15:01.000000000 +0100
+++ panda3d-webgl-port.em/makepanda/makepandacore.py	2019-02-04 21:26:45.671552352 +0100
@@ -2188,7 +2188,7 @@
     if CrossCompiling():
         # We need a version of Python we can run.
         SDK["PYTHONEXEC"] = sys.executable
-        host_version = "python" + sysconfig.get_python_version() + abiflags
+        host_version = "python" + sysconfig.get_python_version()
         if SDK["PYTHONVERSION"] != host_version:
             exit("Host Python version (%s) must be the same as target Python version (%s)!" % (host_version, SDK["PYTHONVERSION"]))
 
@@ -2902,6 +2902,8 @@
 
         null.close()
 
+        null.close()
+
         # Print out the search paths
         if GetVerbose():
             print("System library search path:")
@@ -3278,15 +3280,16 @@
     ORIG_EXT[x] = v
 
 def GetExtensionSuffix():
+    target = GetTarget()
+    if target == 'emscripten':
+        return '.bc'
+
     if sys.version_info >= (3, 0):
         import _imp
         return _imp.extension_suffixes()[0]
 
-    target = GetTarget()
     if target == 'windows':
         return '.pyd'
-    elif target == 'emscripten':
-        return '.bc'
     else:
         return '.so'
 
