--- panda3d-webgl-port/makepanda/makepanda.py	2019-01-06 14:15:01.000000000 +0100
+++ panda3d-webgl-port.em/makepanda/makepanda.py	2019-02-04 21:26:45.679552526 +0100
@@ -28,6 +28,8 @@
 import os
 import sys
 
+if os.environ.get('PYTHONHOME'):del os.environ['PYTHONHOME']
+
 ########################################################################
 ##
 ## PARSING THE COMMAND LINE OPTIONS
@@ -1421,7 +1423,7 @@
                 cmd += ' -mfpu=neon'
 
         elif GetTarget() == 'emscripten':
-            cmd += " -s WARN_ON_UNDEFINED_SYMBOLS=1 -s NO_FILESYSTEM=1"
+            cmd += " -s WARN_ON_UNDEFINED_SYMBOLS=1 -s NO_FILESYSTEM=1 --memory-init-file 0"
 
             if 'VORBIS' in opts and not PkgSkip("VORBIS"):
                 cmd += " -s USE_VORBIS=1"
@@ -1932,7 +1934,7 @@
             cmd += ' -lc -lm'
 
         elif GetTarget() == 'emscripten':
-            cmd += " -s WARN_ON_UNDEFINED_SYMBOLS=1"
+            cmd += " -s WARN_ON_UNDEFINED_SYMBOLS=0"
             if GetOrigExt(dll) == ".exe":
                 cmd += " --memory-init-file 0"
 
@@ -6745,7 +6747,7 @@
 #
 # DIRECTORY: pandatool/src/deploy-stub
 #
-if PkgSkip("PYTHON") == 0:
+if 0:#PkgSkip("PYTHON") == 0:
     OPTS=['DIR:pandatool/src/deploy-stub', 'BUILDING:DEPLOYSTUB']
     PyTargetAdd('deploy-stub.obj', opts=OPTS, input='deploy-stub.c')
     if GetTarget() == 'windows':
