--- panda3d-webgl-port/panda/src/webgldisplay/webGLGraphicsWindow.cxx	2019-01-06 14:15:01.000000000 +0100
+++ panda3d-webgl-port.em/panda/src/webgldisplay/webGLGraphicsWindow.cxx	2019-02-03 23:05:33.257520670 +0100
@@ -169,7 +169,8 @@
   GraphicsWindow::set_properties_now(properties);
 
   if (properties.has_size()) {
-    emscripten_set_canvas_size(properties.get_x_size(), properties.get_y_size());
+    //emscripten_set_canvas_size(properties.get_x_size(), properties.get_y_size());
+	emscripten_set_canvas_element_size("#canvas", _properties.get_x_size(), _properties.get_y_size());
     _properties.set_size(properties.get_size());
     properties.clear_size();
     set_size_and_recalc(_properties.get_x_size(), _properties.get_y_size());
@@ -302,10 +303,18 @@
   }
 
   if (_properties.has_size() && _properties.get_size() != LVecBase2i(1, 1)) {
-    emscripten_set_canvas_size(_properties.get_x_size(), _properties.get_y_size());
+    //emscripten_set_canvas_size(_properties.get_x_size(), _properties.get_y_size());
+	emscripten_set_canvas_element_size("#canvas", _properties.get_x_size(), _properties.get_y_size());
   } else {
-    int width, height, fullscreen;
-    emscripten_get_canvas_size(&width, &height, &fullscreen);
+    int width, height;
+	int fullscreen=0;
+	emscripten_get_canvas_element_size("#canvas", &width, &height);
+	EmscriptenFullscreenChangeEvent fullscreenStatus;
+	EMSCRIPTEN_RESULT er;
+	er = emscripten_get_fullscreen_status(&fullscreenStatus);
+	if (er==EMSCRIPTEN_RESULT_SUCCESS)
+		fullscreen=(int)fullscreenStatus.isFullscreen;
+	
     _properties.set_size(width, height);
     _properties.set_fullscreen(fullscreen > 0);
   }
@@ -464,10 +473,16 @@
     // it does the right thing.  We grab the first unicode code point.
     // Unfortunately, this doesn't seem to handle dead keys on Firefox.
     int keycode = 0;
-    EM_ASM_({
+#ifdef __NOWASM__
+ #if __NOWASM__
+	#error emasm 
+/*    EM_ASM_({
       stringToUTF32(String.fromCharCode($0), $1, 4);
-    }, event->charCode, &keycode);
-
+    }, event->charCode, &keycode);*/
+ #endif
+#else
+	#pragma message "============= __NOWASM__ is not defined assuming WASM=1 ==============="
+#endif
     if (keycode != 0) {
       device->keystroke(keycode);
       return true;
@@ -541,8 +556,15 @@
   case EMSCRIPTEN_EVENT_MOUSEDOWN:
     // Don't register out-of-bounds mouse downs.
     if (event->canvasX >= 0 && event->canvasY >= 0) {
-      int w, h, f;
-      emscripten_get_canvas_size(&w, &h, &f);
+      int w, h;
+	  int f=0;
+	  emscripten_get_canvas_element_size("#canvas", &w, &h);
+ 	  EmscriptenFullscreenChangeEvent fullscreenStatus;
+      EMSCRIPTEN_RESULT er;
+      er = emscripten_get_fullscreen_status(&fullscreenStatus);
+	  if (er==EMSCRIPTEN_RESULT_SUCCESS)
+		f=fullscreenStatus.isFullscreen;
+
       if (event->canvasX < w && event->canvasY < h) {
         device->button_down(MouseButton::button(event->button), time);
         return true;
