--- panda3d-webgl-port/panda/src/express/virtualFileSystem.cxx	2019-01-06 14:15:01.000000000 +0100
+++ panda3d-webgl-port.em/panda/src/express/virtualFileSystem.cxx	2019-02-04 21:26:45.687552700 +0100
@@ -1156,6 +1156,28 @@
   // Also transparently look for a regular file suffixed .pz.
   Filename strpath_pz = strpath + ".pz";
 
+//PMPP
+#ifdef __NOWASM__
+ #if __NOWASM__
+    #ifdef CPPPARSER
+        #warning vfs hack at virtualFileSystem.cxx:1155
+    #else
+        if ( strpath_pz.exists() ){
+            // compressed file already here
+        } else {
+            if ( strpath.exists() ){
+                // raw file is already here
+            } else {
+                int rx = EM_ASM_INT( {  return window.callfs( Pointer_stringify($0) ); }, strpath.c_str() );
+            }
+        }
+    #endif
+ #endif
+#else
+	#pragma message "============= __NOWASM__ is not defined assuming WASM=1 ==============="
+#endif
+//PMPP
+
   // Now scan all the mount points, from the back (since later mounts override
   // more recent ones), until a match is found.
   PT(VirtualFile) found_file = nullptr;
