--- panda3d-webgl-port/panda/src/pipeline/threadSimpleManager.cxx	2018-05-07 17:07:51.000000000 +0200
+++ panda3d-webgl-port.pp/panda/src/pipeline/threadSimpleManager.cxx	2018-10-27 21:38:09.006635046 +0200
@@ -472,9 +472,10 @@
 #ifdef HAVE_PYTHON
     // Ensure that the Python threading system is initialized and ready to go.
 
-#if PY_VERSION_HEX >= 0x03020000
-    Py_Initialize();
-#endif
+// see https://github.com/pmp-p/pydk/blob/master/sources.32/panda3d-input-overhaul.patchset/panda_src_putil_pythonCallbackObject.cxx.diff
+//#if PY_VERSION_HEX >= 0x03020000
+//    Py_Initialize();
+//#endif
 
     PyEval_InitThreads();
 #endif
