--- panda3d-webgl-port/panda/src/express/trueClock.cxx	2018-05-07 17:07:51.000000000 +0200
+++ panda3d-webgl-port.pp/panda/src/express/trueClock.cxx	2018-10-27 21:37:58.403635292 +0200
@@ -480,47 +480,7 @@
   _time_scale = new_time_scale;
 }
 
-#elif defined(__EMSCRIPTEN__)
-
-/**
- * The Emscripten implementation.  This uses either the JavaScript function
- * performance.now() if available, otherwise Date.now().
- */
-
-#include <emscripten.h>
-
-/**
- *
- */
-double TrueClock::
-get_long_time() {
-  return emscripten_get_now() * 0.001;
-}
-
-/**
- *
- */
-double TrueClock::
-get_short_raw_time() {
-  return emscripten_get_now() * 0.001;
-}
-
-/**
- *
- */
-bool TrueClock::
-set_cpu_affinity(uint32_t mask) const {
-  return false;
-}
-
-/**
- *
- */
-TrueClock::
-TrueClock() {
-}
-
-#else  // !WIN32_VC
+#else
 
 // The Posix implementation.
 
