--- Python-3.8.0b4/Modules/mmapmodule.c	2019-08-29 23:59:20.000000000 +0200
+++ Python-3.8.0b4.32/Modules/mmapmodule.c	2019-08-31 11:57:53.541672005 +0200
@@ -79,6 +79,12 @@
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    extern void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
+#endif
+//PMPP API<21
+
 typedef enum
 {
     ACCESS_DEFAULT,
