--- Python-3.8.0b4/Python/preconfig.c	2019-08-29 23:59:20.000000000 +0200
+++ Python-3.8.0b4.32/Python/preconfig.c	2019-08-31 11:57:53.581670474 +0200
@@ -788,7 +788,10 @@
     }
 
     preconfig_get_global_vars(config);
-
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+    #define BADLOC 1
+#else
+    #define BADLOC 0
     /* Copy LC_CTYPE locale, since it's modified later */
     const char *loc = setlocale(LC_CTYPE, NULL);
     if (loc == NULL) {
@@ -798,6 +801,8 @@
     if (init_ctype_locale == NULL) {
         return _PyStatus_NO_MEMORY();
     }
+#endif
+
 
     /* Save the config to be able to restore it if encodings change */
     PyPreConfig save_config;
@@ -894,10 +899,13 @@
     status = _PyStatus_OK();
 
 done:
+#if BADLOC
+#else
     if (init_ctype_locale != NULL) {
         setlocale(LC_CTYPE, init_ctype_locale);
         PyMem_RawFree(init_ctype_locale);
     }
+#endif
     Py_UTF8Mode = init_utf8_mode ;
 #ifdef MS_WINDOWS
     Py_LegacyWindowsFSEncodingFlag = init_legacy_encoding;
